<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-05-20" />

<title>FibroblastAnalysisDEgenes_wtonly_withoutGlial</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cxcl13Hpb</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FibroblastAnalysisDEgenes_wtonly_withoutGlial</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-05-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cxcl13Hpb/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240311code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240311code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240311)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allFb_naiveANDd4_seurat.rds
</td>
<td style="text-align:left;">
data/merged_allFb_naiveANDd4_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/DEGenesFbcond3_wtonly_noglial.txt
</td>
<td style="text-align:left;">
data/DEGenesFbcond3_wtonly_noglial.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree49708743e7a501b3b2a0d6c97ca872d1488e8064targetblank4970874a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/49708743e7a501b3b2a0d6c97ca872d1488e8064" target="_blank">4970874</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree49708743e7a501b3b2a0d6c97ca872d1488e8064targetblank4970874a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/49708743e7a501b3b2a0d6c97ca872d1488e8064" target="_blank">4970874</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/dataJuli2021/
    Ignored:    data/dataJuli2022/
    Ignored:    data/dataJune2019/
    Ignored:    data/dataSept2020/

Untracked files:
    Untracked:  VennDiagram.2025-06-04_10-41-23.549207.log
    Untracked:  VennDiagram.2025-06-04_10-41-23.686772.log
    Untracked:  analysis/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd
    Untracked:  venn-4-diagram.png
    Untracked:  venn-4-diagram.png.2025-06-04_10-41-23.292887.log

Unstaged changes:
    Deleted:    analysis/DEGenesD4cond2.txt
    Deleted:    analysis/DEGenesD4cond3.txt
    Deleted:    analysis/DEGenesFbGrancond2.txt
    Deleted:    analysis/DEGenesFbcond2.txt
    Deleted:    analysis/DEGenesFbcond3.txt
    Deleted:    analysis/DEGenesFbcond3_wtonly.txt
    Modified:   analysis/FRCsignatures.Rmd
    Deleted:    analysis/FbDEGenescond3.txt
    Deleted:    analysis/FbmarkerGenesclustername.txt
    Deleted:    analysis/FbmarkerGenesclustername_wtonly.txt
    Modified:   analysis/FibroblastAnalysisDEgenes.Rmd
    Modified:   analysis/FibroblastAnalysisDEgenes_wtonly.Rmd
    Modified:   analysis/FibroblastAnalysis_wtonly.Rmd
    Deleted:    analysis/GranDEGenescond2.txt
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)</code></pre>
</div>
<div id="load-object-merged-allfb-d4" class="section level2">
<h2>load object merged allFb d4</h2>
<pre class="r"><code>##load object merged allFb naive and d4
fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allFb_naiveANDd4_seurat.rds&quot;
seuratFb &lt;- readRDS(fileNam)
table(seuratFb$dataset)</code></pre>
<pre><code>
                10_20190604_Mouse_LP_Cxcl13cre_IL33flfl_EYFP_v3 
                                                           6591 
                       10_20200923_Mu_LP_Cxcl13Cre_EYFP_HpD4_v3 
                                                           7409 
        10_20210624_Mu_granuloma_Cxcl13Cre_IL33flf_EYFP_HpD4_v3 
                                                            627 
                         11_20190604_Mouse_LP_Cxcl13cre_EYFP_v3 
                                                           4561 
                         12_20190604_Mouse_LP_Cxcl13cre_EYFP_v3 
                                                           7301 
               7_20200923_Mu_LP_Cxcl13Cre_IL33flfl_EYFP_HpD4_v3 
                                                           7816 
               8_20200923_Mu_LP_Cxcl13Cre_IL33fllf_EYFP_HpD4_v3 
                                                           5899 
                 9_20190604_Mouse_LP_Cxcl13cre_IL33flfl_EYFP_v3 
                                                           6792 
                        9_20200923_Mu_LP_Cxcl13Cre_EYFP_HpD4_v3 
                                                           8550 
                 9_20210624_Mu_granuloma_Cxcl13Cre_EYFP_HpD4_v3 
                                                           1177 
       o289411_1-1_20220629_Mu_granuloma_Cxcl13Cre_CD45_HpD4_v3 
                                                              4 
        o289411_2-2_20220629_Mu_granuloma_Cxcl13Cre_FSC_HpD4_v3 
                                                           2319 
o289411_3-3_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_CD45_HpD4_v3 
                                                              7 
 o289411_4-4_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_FSC_HpD4_v3 
                                                           5165 </code></pre>
</div>
<div id="subset-wt" class="section level2">
<h2>subset wt</h2>
<pre class="r"><code>table(seuratFb$cond2)</code></pre>
<pre><code>
   WT   cko 
31321 32897 </code></pre>
<pre class="r"><code>seuratwt &lt;- subset(seuratFb, cond2 == &quot;WT&quot;)
table(seuratwt$cond2)</code></pre>
<pre><code>
   WT 
31321 </code></pre>
<pre class="r"><code>table(seuratwt$clustername)</code></pre>
<pre><code>
PdgfraloFb1 PdgfraloFb2 PdgfraloFb3 PdgfraloFb4 Trophocytes   Telocytes    Myocytes    Pdgfrahi 
      11352        7278        2552         403        6582        1701         756         523 
     Thy1Fb       Glial 
        111          63 </code></pre>
<pre class="r"><code>seuratnoglial &lt;- subset(seuratwt, clustername == &quot;Glial&quot;, invert = TRUE)
table(seuratnoglial$clustername)</code></pre>
<pre><code>
PdgfraloFb1 PdgfraloFb2 PdgfraloFb3 PdgfraloFb4 Trophocytes   Telocytes    Myocytes    Pdgfrahi 
      11352        7278        2552         403        6582        1701         756         523 
     Thy1Fb 
        111 </code></pre>
<pre class="r"><code>seuratFb &lt;- seuratnoglial</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color vectors</h2>
<pre class="r"><code>## set vector for cond2
colcond2 &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;)
names(colcond2) &lt;- c(&quot;WT&quot;, &quot;cko&quot;)

## set vector for cond2
colcond3 &lt;- c(&quot;#B45B5C&quot;,&quot;#628395&quot;,&quot;#BF782D&quot;)
names(colcond3) &lt;- c(&quot;infectedD4&quot;, &quot;naive&quot;, &quot;granulomaD4&quot;)

#colPal &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#779d8d&quot;,&quot;#0F1F38&quot;,&quot;#E84A5F&quot;,&quot;#FF847C&quot;,&quot;#F8B195&quot;, &quot;#727077&quot;, &quot;#C06C84&quot;,&quot;#2A363B&quot;, &quot;#6C5B7B&quot;)
#names(colPal) &lt;- c(&quot;0&quot; ,&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;)

colPal &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#99B898&quot;,&quot;#E84A5F&quot;,&quot;#0F1F38&quot;,&quot;#FF847C&quot;,&quot;#F8B195&quot;,&quot;#6C5B7B&quot;, &quot;#C06C84&quot;, &quot;#727077&quot;)
names(colPal) &lt;- c(&quot;0&quot; ,&quot;1&quot;, &quot;2&quot;, &quot;3&quot;,&quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;)

colclustername &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#99B898&quot;,&quot;#E84A5F&quot;,&quot;#0F1F38&quot;,&quot;#F8B195&quot;,&quot;#727077&quot;,&quot;#FF847C&quot;,&quot;#C06C84&quot;,&quot;#904D39&quot;)
names(colclustername) &lt;- c(&quot;PdgfraloFb1&quot; ,&quot;PdgfraloFb2&quot;, &quot;Trophocytes&quot;,&quot;PdgfraloFb3&quot;, &quot;Telocytes&quot;,&quot;Myocytes&quot;, &quot;Pdgfrahi&quot;, &quot;PdgfraloFb4&quot;, &quot;Thy1Fb&quot;, &quot;Glial&quot;)</code></pre>
</div>
<div id="calculate-de-genes-according-to-cond3" class="section level2">
<h2>calculate DE genes according to cond3</h2>
<pre class="r"><code>## calculate DE genes
Idents(seuratFb) &lt;- seuratFb$cond3
levels(seuratFb)

DEGenesFbcond3 &lt;- FindAllMarkers(seuratFb, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) 

write.table(DEGenesFbcond3, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/DEGenesFbcond3_wtonly_noglial.txt&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="gsea-fb-gran" class="section level2">
<h2>GSEA Fb gran</h2>
<pre class="r"><code>## load DEGenesFbcond3
DEGenesFbcond3 &lt;- read.delim(&quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/DEGenesFbcond3_wtonly_noglial.txt&quot;, header = TRUE, sep = &quot;\t&quot;)

## GSEA on genes upregulated in granulomaFb
##adjust table
DEGenesGran &lt;- dplyr::filter(DEGenesFbcond3, cluster == &quot;granulomaD4&quot;)
DEGenesGran &lt;- DEGenesGran %&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA Gran
egoGran &lt;- enrichGO(gene = unique(DEGenesGran$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoGran &lt;- setReadable(egoGran, OrgDb = org.Mm.eg.db)
dotplot(egoGran, showCategory=30)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/GSEA%20Fb%20gran-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-gsea-gran" class="section level2">
<h2>barplot GSEA gran</h2>
<pre class="r"><code>#select pahtways and filter file
selGran &lt;- c(&quot;leukocyte migration&quot;,&quot;regulation of innate immune response&quot;, &quot;mitochondrial translation&quot;, &quot;response to interferon-beta&quot;)
egoGran_fil &lt;- egoGran%&gt;% filter(egoGran@result$Description %in% selGran)

egoGran_fil_2 &lt;- pairwise_termsim(egoGran_fil)
#make ggbarplot
p &lt;- ggbarplot(egoGran_fil_2@result, x = &quot;Description&quot;, y = &quot;Count&quot;, fill = &quot;#BF782D&quot;, color= &quot;#BF782D&quot;, sort.val = &quot;asc&quot;, orientation = &quot;horizontal&quot;)
p</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/barplot%20GSEA%20Fb%20gran-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="convert-to-sce" class="section level2">
<h2>convert to sce</h2>
<pre class="r"><code>##convert seurat object to sce object
sce &lt;- as.SingleCellExperiment(seuratFb)
genes &lt;- data.frame(geneID=rownames(sce)) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;, &quot;#f7f7f7&quot;, &quot;#f4a582&quot;, &quot;#b2183c&quot;, &quot;#85122d&quot;))

##subset naive and convert to sce
seuratFbnaive &lt;- subset(seuratFb, cond3 == &quot;naive&quot;)
DimPlot(seuratFbnaive, reduction= &quot;umap&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/convert%20to%20sce-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sceFbnaive &lt;- as.SingleCellExperiment(seuratFbnaive)

##subset SI and convert to sce
seuratFbinf &lt;- subset(seuratFb, cond3 == &quot;infectedD4&quot;)
DimPlot(seuratFbinf, reduction= &quot;umap&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/convert%20to%20sce-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sceFbinf &lt;- as.SingleCellExperiment(seuratFbinf)

##subset Gran and convert to sce
seuratFbgran &lt;- subset(seuratFb, cond3 == &quot;granulomaD4&quot;)
DimPlot(seuratFbgran, reduction= &quot;umap&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/convert%20to%20sce-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sceFbgran &lt;- as.SingleCellExperiment(seuratFbgran)</code></pre>
</div>
<div id="plot-gsea-signatures-onto-umaps-gran" class="section level2">
<h2>plot GSEA signatures onto umaps gran</h2>
<pre class="r"><code>## GSEA on genes upregulated in granuloma
ego1 &lt;- dplyr::filter(egoGran@result,egoGran@result$Description==&quot;leukocyte migration&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.6))
sceSub$sign2[which(sceSub$sign &gt; 0.6)] &lt;- 0.6
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.6336156</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.6))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 0.6)] &lt;- 0.6
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 0.4226704</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.6))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 0.6)] &lt;- 0.6
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 0.5133976</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.6))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 0.6)] &lt;- 0.6
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 0.6336156</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20leukocyte%20migration-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## GSEA on genes upregulated in granuloma
ego1 &lt;- dplyr::filter(egoGran@result,egoGran@result$Description==&quot;regulation of innate immune response&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.7))
sceSub$sign2[which(sceSub$sign &gt; 0.7)] &lt;- 0.7
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.7604629</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.7))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 0.7)] &lt;- 0.7
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 0.7604629</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.7))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 0.7)] &lt;- 0.7
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 0.7094693</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.7))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 0.7)] &lt;- 0.7
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 0.7341736</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20positive%20regualtion%20of%20innate%20immune%20response-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-fb-si" class="section level2">
<h2>GSEA Fb SI</h2>
<pre class="r"><code>## GSEA on genes upregulated in SI
##adjust table
DEGenesinf &lt;- dplyr::filter(DEGenesFbcond3, cluster == &quot;infectedD4&quot;)
DEGenesinf &lt;- DEGenesinf %&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA inf
egoinf &lt;- enrichGO(gene = unique(DEGenesinf$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoinf &lt;- setReadable(egoinf, OrgDb = org.Mm.eg.db)
dotplot(egoinf, showCategory=30)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/GSEA%20Fb%20SI-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-gsea-si" class="section level2">
<h2>barplot GSEA SI</h2>
<pre class="r"><code>#select pahtways and filter file
selSI &lt;- c(&quot;extracellular matrix organization&quot;,&quot;regulation of cellular response to growth factor stimulus&quot;, &quot;regulation of Wnt signaling pathway&quot;, &quot;response to virus&quot;)
egoSI_fil &lt;- egoinf%&gt;% filter(egoinf@result$Description %in% selSI)

egoSI_fil_2 &lt;- pairwise_termsim(egoSI_fil)
#make ggbarplot
p &lt;- ggbarplot(egoSI_fil_2@result, x = &quot;Description&quot;, y = &quot;Count&quot;, fill = &quot;#B45B5C&quot;, color= &quot;#B45B5C&quot;, sort.val = &quot;asc&quot;, orientation = &quot;horizontal&quot;)
p</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/barplot%20GSEA%20Fb%20SI-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-gsea-signatures-onto-umaps-inf" class="section level2">
<h2>plot GSEA signatures onto umaps inf</h2>
<pre class="r"><code>## GSEA on genes upregulated in inf
ego1 &lt;- dplyr::filter(egoinf@result,egoinf@result$Description==&quot;extracellular matrix organization&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.4))
sceSub$sign2[which(sceSub$sign &gt; 1.4)] &lt;- 1.4
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 1.447404</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.4))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 1.4)] &lt;- 1.4
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 1.295443</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.4))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 1.4)] &lt;- 1.4
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 1.422564</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.4))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 1.4)] &lt;- 1.4
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 1.447404</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20extracellular%20matrix%20organization-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## GSEA on genes upregulated in inf
ego1 &lt;- dplyr::filter(egoinf@result,egoinf@result$Description==&quot;regulation of cellular response to growth factor stimulus&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.2))
sceSub$sign2[which(sceSub$sign &gt; 1.2)] &lt;- 1.2
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 1.297237</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.2))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 1.2)] &lt;- 1.2
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 1.190929</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.2))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 1.2)] &lt;- 1.2
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 1.297237</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1.2))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 1.2)] &lt;- 1.2
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 1.158798</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20regulation%20of%20cellular%20response%20to%20growth%20factor%20stimulus-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-fb-naive" class="section level2">
<h2>GSEA Fb naive</h2>
<pre class="r"><code>## GSEA on genes upregulated in naive
##adjust table
DEGenesnaive &lt;- dplyr::filter(DEGenesFbcond3, cluster == &quot;naive&quot;)
DEGenesnaive &lt;- DEGenesnaive %&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA inf
egonaive &lt;- enrichGO(gene = unique(DEGenesnaive$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egonaive &lt;- setReadable(egonaive, OrgDb = org.Mm.eg.db)
dotplot(egonaive, showCategory=30)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/GSEA%20Fb%20naive-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-gsea-naive" class="section level2">
<h2>barplot GSEA naive</h2>
<pre class="r"><code>#select pahtways and filter file
selnaive &lt;- c(&quot;negative regulation of phosphate metabolic process&quot;, &quot;response to steroid hormone&quot;, &quot;G1/S transition of mitotic cell cycle&quot;, &quot;regulation of developmental growth&quot;)
egonaive_fil &lt;- egonaive%&gt;% filter(egonaive@result$Description %in% selnaive)

egonaive_fil_2 &lt;- pairwise_termsim(egonaive_fil)
#make ggbarplot
p &lt;- ggbarplot(egonaive_fil_2@result, x = &quot;Description&quot;, y = &quot;Count&quot;, fill = &quot;#628395&quot;, color= &quot;#628395&quot;, sort.val = &quot;asc&quot;, orientation = &quot;horizontal&quot;)
p</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/barplot%20GSEA%20Fb%20naive-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-gsea-signatures-onto-umaps-naive" class="section level2">
<h2>plot GSEA signatures onto umaps naive</h2>
<pre class="r"><code>## GSEA on genes upregulated in naive
ego1 &lt;- dplyr::filter(egonaive@result,egonaive@result$Description ==&quot;negative regulation of phosphate metabolic process&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceSub$sign2[which(sceSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.936732</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 0.936732</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 0.8262298</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 0.8069812</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20negative%20regulation%20of%20phosphate%20metabolic%20process-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## GSEA on genes upregulated in naive
ego1 &lt;- dplyr::filter(egonaive@result,egonaive@result$Description ==&quot;response to steroid hormone&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceSub$sign2[which(sceSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.98905</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 0.98905</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 0.8834707</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 0.8098008</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20response%20to%20steroid%20hormone-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## GSEA on genes upregulated in naive
ego1 &lt;- dplyr::filter(egonaive@result,egonaive@result$Description ==&quot;G1/S transition of mitotic cell cycle&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceSub$sign2[which(sceSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.9581044</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes naive
sceFbnaiveSub &lt;- sceFbnaive[which(rownames(sceFbnaive) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbnaiveSub@assays@data$logcounts)))/nrow(signGenes)
sceFbnaiveSub$sign &lt;- cntMat
sceFbnaiveSub$sign2 &lt;- sceFbnaiveSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbnaiveSub$sign2[which(sceFbnaiveSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbnaiveSub$sign)</code></pre>
<pre><code>[1] 0.9581044</code></pre>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbnaiveSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes inf
sceFbinfSub &lt;- sceFbinf[which(rownames(sceFbinf) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbinfSub@assays@data$logcounts)))/nrow(signGenes)
sceFbinfSub$sign &lt;- cntMat
sceFbinfSub$sign2 &lt;- sceFbinfSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbinfSub$sign2[which(sceFbinfSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbinfSub$sign)</code></pre>
<pre><code>[1] 0.9460293</code></pre>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbinfSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make a count matrix of signature genes gran
sceFbgranSub &lt;- sceFbgran[which(rownames(sceFbgran) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceFbgranSub@assays@data$logcounts)))/nrow(signGenes)
sceFbgranSub$sign &lt;- cntMat
sceFbgranSub$sign2 &lt;- sceFbgranSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.9))
sceFbgranSub$sign2[which(sceFbgranSub$sign &gt; 0.9)] &lt;- 0.9
##check max and min values
max(sceFbgranSub$sign)</code></pre>
<pre><code>[1] 0.6667127</code></pre>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceFbgranSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd/plot%20GSEA%20G1/S%20transition%20of%20mitotic%20cell%20cycle-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Jun  4 10:53:24 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.1              
 [4] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
 [7] msigdbr_10.0.2              org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.12                 
[13] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.4             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.4                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.5.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.2               Seurat_5.3.0               
[43] SeuratObject_5.1.0          sp_2.2-0                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] data.table_1.17.2         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          RColorBrewer_1.1-3        progressr_0.15.1         
 [91] tweenr_2.0.3              later_1.4.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.6              pkgconfig_2.0.3           TMB_1.9.17               
[103] spatstat.univar_3.1-3     mathjaxr_1.8-0            EnvStats_3.1.0           
[106] aplot_0.2.5               scatterplot3d_0.3-44      ape_5.8-1                
[109] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[112] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[115] tools_4.4.0               globals_0.18.0            beeswarm_0.4.0           
[118] broom_1.0.8               nlme_3.1-168              lambda.r_1.2.4           
[121] assertthat_0.2.1          lme4_1.1-37               digest_0.6.37            
[124] numDeriv_2016.8-1.1       Matrix_1.7-3              farver_2.1.2             
[127] tzdb_0.5.0                remaCor_0.0.18            reshape2_1.4.4           
[130] yulab.utils_0.2.0         glue_1.8.0                cachem_1.1.0             
[133] polyclip_1.10-7           generics_0.1.4            Biostrings_2.72.1        
[136] mvtnorm_1.3-3             parallelly_1.44.0         mnormt_2.1.1             
[139] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[142] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[145] httr2_1.1.2               spam_2.11-1               gson_0.1.0               
[148] graphlayouts_1.2.2        gtools_3.9.5              ggsignif_0.6.4           
[151] RcppEigen_0.3.4.0.2       shiny_1.10.0              GenomeInfoDbData_1.2.12  
[154] glmmTMB_1.1.11            R.utils_2.13.0            memoise_2.0.1            
[157] rmarkdown_2.29            scales_1.4.0              R.methodsS3_1.8.2        
[160] future_1.49.0             RANN_2.6.2                Cairo_1.6-2              
[163] spatstat.data_3.1-6       rstudioapi_0.17.1         cluster_2.1.8.1          
[166] mutoss_0.1-13             spatstat.utils_3.1-3      hms_1.1.3                
[169] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[172] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[175] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[178] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[181] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[184] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[187] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[190] Rdpack_2.6.4              promises_1.3.2            scatterpie_0.2.4         
[193] RSQLite_2.3.11            qvalue_2.36.0             sandwich_3.1-1           
[196] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[199] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[202] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[205] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[208] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[211] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[214] spatstat.random_3.3-3     R6_2.6.1                  fastmap_1.2.0            
[217] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[220] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[223] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[226] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[229] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[232] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-2   
[235] lifecycle_1.0.4           blme_1.0-6                msigdbdf_24.1.1          
[238] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[241] robustbase_0.99-4-1       spatstat.geom_3.3-6       sn_2.1.1                 
[244] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[247] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[250] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.1              
 [4] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
 [7] msigdbr_10.0.2              org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.12                 
[13] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.4             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.4                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.5.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.2               Seurat_5.3.0               
[43] SeuratObject_5.1.0          sp_2.2-0                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] data.table_1.17.2         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          RColorBrewer_1.1-3        progressr_0.15.1         
 [91] tweenr_2.0.3              later_1.4.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.6              pkgconfig_2.0.3           TMB_1.9.17               
[103] spatstat.univar_3.1-3     mathjaxr_1.8-0            EnvStats_3.1.0           
[106] aplot_0.2.5               scatterplot3d_0.3-44      ape_5.8-1                
[109] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[112] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[115] tools_4.4.0               globals_0.18.0            beeswarm_0.4.0           
[118] broom_1.0.8               nlme_3.1-168              lambda.r_1.2.4           
[121] assertthat_0.2.1          lme4_1.1-37               digest_0.6.37            
[124] numDeriv_2016.8-1.1       Matrix_1.7-3              farver_2.1.2             
[127] tzdb_0.5.0                remaCor_0.0.18            reshape2_1.4.4           
[130] yulab.utils_0.2.0         glue_1.8.0                cachem_1.1.0             
[133] polyclip_1.10-7           generics_0.1.4            Biostrings_2.72.1        
[136] mvtnorm_1.3-3             parallelly_1.44.0         mnormt_2.1.1             
[139] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[142] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[145] httr2_1.1.2               spam_2.11-1               gson_0.1.0               
[148] graphlayouts_1.2.2        gtools_3.9.5              ggsignif_0.6.4           
[151] RcppEigen_0.3.4.0.2       shiny_1.10.0              GenomeInfoDbData_1.2.12  
[154] glmmTMB_1.1.11            R.utils_2.13.0            memoise_2.0.1            
[157] rmarkdown_2.29            scales_1.4.0              R.methodsS3_1.8.2        
[160] future_1.49.0             RANN_2.6.2                Cairo_1.6-2              
[163] spatstat.data_3.1-6       rstudioapi_0.17.1         cluster_2.1.8.1          
[166] mutoss_0.1-13             spatstat.utils_3.1-3      hms_1.1.3                
[169] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[172] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[175] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[178] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[181] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[184] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[187] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[190] Rdpack_2.6.4              promises_1.3.2            scatterpie_0.2.4         
[193] RSQLite_2.3.11            qvalue_2.36.0             sandwich_3.1-1           
[196] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[199] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[202] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[205] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[208] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[211] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[214] spatstat.random_3.3-3     R6_2.6.1                  fastmap_1.2.0            
[217] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[220] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[223] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[226] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[229] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[232] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-2   
[235] lifecycle_1.0.4           blme_1.0-6                msigdbdf_24.1.1          
[238] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[241] robustbase_0.99-4-1       spatstat.geom_3.3-6       sn_2.1.1                 
[244] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[247] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[250] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
