<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2024-06-12" />

<title>GranulomaMacrophages</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cxcl13Hpb</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GranulomaMacrophages</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2024-06-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cxcl13Hpb/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240311code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240311code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240311)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allgranuloma_wofilter_seurat_wocl21.rds
</td>
<td style="text-align:left;">
data/merged_allgranuloma_wofilter_seurat_wocl21.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/analysis/DEgenesMph_cluster_name
</td>
<td style="text-align:left;">
analysis/DEgenesMph_cluster_name
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/analysis/DEgenesMph1vsMph3
</td>
<td style="text-align:left;">
analysis/DEgenesMph1vsMph3
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree3d5dd42d4a98d26681a2c95ebf9d626ca1ca4647targetblank3d5dd42a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/3d5dd42d4a98d26681a2c95ebf9d626ca1ca4647" target="_blank">3d5dd42</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree3d5dd42d4a98d26681a2c95ebf9d626ca1ca4647targetblank3d5dd42a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/3d5dd42d4a98d26681a2c95ebf9d626ca1ca4647" target="_blank">3d5dd42</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/dataJuli2021/
    Ignored:    data/dataJuli2022/
    Ignored:    data/dataJune2019/
    Ignored:    data/dataSept2020/

Untracked files:
    Untracked:  analysis/DEgenesMph1vsMph3
    Untracked:  analysis/DEgenesMph_cluster_name
    Untracked:  analysis/FbmarkerGenesclustername.txt
    Untracked:  analysis/FibroblastAnalysis.Rmd
    Untracked:  analysis/GranulomaAnalysis.Rmd
    Untracked:  analysis/GranulomaMacrophages.Rmd
    Untracked:  analysis/markergenes_cluster_name
    Untracked:  analysis/vorlagen.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)</code></pre>
</div>
<div id="load-merged-file-granuloma" class="section level2">
<h2>load merged file granuloma</h2>
<pre class="r"><code>##load merged file 
fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allgranuloma_wofilter_seurat_wocl21.rds&quot;
seurat &lt;- readRDS(fileNam)
table(seurat$dataset)</code></pre>
<pre><code>
        10_20210624_Mu_granuloma_Cxcl13Cre_IL33flf_EYFP_HpD4_v3 
                                                           5988 
                 9_20210624_Mu_granuloma_Cxcl13Cre_EYFP_HpD4_v3 
                                                           4278 
       o289411_1-1_20220629_Mu_granuloma_Cxcl13Cre_CD45_HpD4_v3 
                                                           8013 
        o289411_2-2_20220629_Mu_granuloma_Cxcl13Cre_FSC_HpD4_v3 
                                                           3008 
o289411_3-3_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_CD45_HpD4_v3 
                                                           6909 
 o289411_4-4_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_FSC_HpD4_v3 
                                                           6472 </code></pre>
<pre class="r"><code>table(seurat$RNA_snn_res.0.25)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19 
4607 4144 3007 2477 2382 2201 2173 2048 1944 1766 1502 1137  870  848  798  766  653  639  357  180 
  20   21 
 169    0 </code></pre>
<pre class="r"><code>table(seurat$orig.ident)</code></pre>
<pre><code>
      
34668 </code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color vectors</h2>
<pre class="r"><code>### col clustername
colclustername &lt;- c(&quot;#355C7D&quot;,&quot;#B1746FFF&quot;,&quot;#202547&quot;,&quot;#B09C85&quot;, &quot;#4e5a4c&quot;,&quot;#53354A&quot;,&quot;#2A363B&quot;,&quot;#8491B4FF&quot;,&quot;#00A087FF&quot;,&quot;#DC9989&quot;,&quot;#84ad83&quot;, &quot;#628395&quot;,&quot;#D33B44&quot;,&quot;#779d8d&quot;,&quot;#727077&quot;,&quot;#868686FF&quot;,&quot;#F8B195&quot;,&quot;#FF847C&quot;,&quot;#725663FF&quot;,&quot;#904D39&quot;,&quot;#91D1C2&quot;)
names(colclustername) &lt;- c(&quot;PdgfraloFb1&quot;,&quot;Mph1&quot;, &quot;CD8Tcell1&quot;,&quot;Neut&quot;, &quot;IgABcell1&quot;, &quot;LEC&quot;, &quot;Telocytes&quot;, &quot;DC&quot;, &quot;CD4Tcell/ILC2&quot;, &quot;Mph2&quot;,  &quot;NKcell&quot;, &quot;CD8Tcell2&quot;, &quot;Mph3&quot;, &quot;Trophocytes&quot;, &quot;PdgfrahiLy6anegFb&quot;, &quot;IgABcell2&quot;, &quot;Myocytes&quot;, &quot;PdgfraloFb2&quot;, &quot;IgDBcell&quot;, &quot;NC&quot;, &quot;ICC&quot;)

### col cond2
colcond2 &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;)
names(colcond2) &lt;- c(&quot;WT&quot;, &quot;cko&quot;)</code></pre>
</div>
<div id="subset-macrophages" class="section level2">
<h2>subset macrophages</h2>
<pre class="r"><code>##subset macrophages
seuratMph &lt;- subset(seurat, cluster_name %in% c(&quot;Mph1&quot;,&quot;Mph2&quot;,&quot;Mph3&quot;))
Idents(seuratMph) &lt;- seuratMph$cluster_name
levels(seuratMph)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph2&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>DimPlot(seuratMph, reduction = &quot;umap&quot;, cols = colclustername) </code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/subset%20mph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-macrophages" class="section level2">
<h2>relative abundance macrophages</h2>
<pre class="r"><code>## Abundance plot Mph
##make count list
datList &lt;- NULL
for(con in unique(seuratMph$cond2)){
  seuratSub &lt;- subset(seuratMph, cond2==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$cluster_name)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(cond2=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 29085  3707
[1] 29085  3073</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

ordXcond2 &lt;- c(&quot;WT&quot;, &quot;cko&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;cond2&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster_name&quot;, ylab = &quot;frequency&quot;, palette = colclustername, order = ordXcond2) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/relative%20abundance%20mph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="de-genes-mph" class="section level2">
<h2>DE genes mph</h2>
<pre class="r"><code>##calculate DE genes 
Idents(seuratMph) &lt;- seuratMph$cluster_name
levels(seuratMph)
DEGenesMph &lt;- FindAllMarkers(seuratMph, only.pos=T, logfc.threshold = 0.2) %&gt;% filter(p_val_adj &lt; 0.01) 

#write.table(DEGenesMph, 
            #file= &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/analysis/DEgenesMph_cluster_name&quot;,
            #sep=&quot;\t&quot;,
            #quote=F,
            #row.names=F,
            #col.names=T)</code></pre>
</div>
<div id="mph1-vs-mph3" class="section level2">
<h2>mph1 vs mph3</h2>
<pre class="r"><code>##filter mph1 and mph3
seuratMph1and3 &lt;- subset(seurat, cluster_name %in% c(&quot;Mph1&quot;,&quot;Mph3&quot;))
Idents(seuratMph1and3) &lt;- seuratMph1and3$cluster_name
levels(seuratMph1and3)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>DimPlot(seuratMph1and3, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclustername) </code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/filter%20mph1%20and%20mph3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="de-genes-mph1-vs-mph3" class="section level2">
<h2>DE genes mph1 vs mph3</h2>
<pre class="r"><code>##calculate DE genes 
Idents(seuratMph1and3) &lt;- seuratMph1and3$cluster_name
levels(seuratMph1and3)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>DEGenesMph1and3 &lt;- FindAllMarkers(seuratMph1and3, only.pos=T, logfc.threshold = 0.2) %&gt;% filter(p_val_adj &lt; 0.01) 

#write.table(DEGenesMph1and3, 
            #file= &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/analysis/DEgenesMph1vsMph3&quot;,
            #sep=&quot;\t&quot;,
            #quote=F,
            #row.names=F,
            #col.names=T)</code></pre>
</div>
<div id="scatterplot-de-genes-mph1-vs-mph3" class="section level2">
<h2>scatterplot DE genes mph1 vs mph3</h2>
<pre class="r"><code>### table with expression levels of all genes
allID &lt;- seuratMph1and3
Idents(allID) &lt;- allID$cluster_name
avg.allID &lt;- AverageExpression(allID, group.by = &quot;cluster_name&quot;)
avg.allID &lt;- data.frame(log1p(avg.allID$RNA)) %&gt;% 
  rownames_to_column(var=&quot;gene&quot;)

## table with genes of a certain expression frequency
GeneFreq &lt;- as.data.frame(GetAssayData(allID, assay = &quot;RNA&quot;, slot = &quot;data&quot;)) %&gt;%
  tibble::rownames_to_column(var=&quot;gene&quot;) %&gt;% mutate(count=rowSums(.!=0)-1) %&gt;%
  mutate(countFreq=count/ncol(allID)) %&gt;%
  dplyr::select(gene, count, countFreq) %&gt;%
  dplyr::filter(countFreq&gt;=0.02)

## table with DEGenes to labeel - avg_log2FC &gt; 1.5 and p_val_adj &lt; 1e-100
genes.to.label &lt;- DEGenesMph1and3 %&gt;% group_by(.,cluster) %&gt;%
  dplyr::filter(p_val_adj &lt; 1e-100) %&gt;% dplyr::filter(avg_log2FC &gt; 1.5) %&gt;% mutate(labelNam=gsub(&quot;^.*\\.&quot;,&quot;&quot;,gene)) 
  
avg.allID &lt;- avg.allID %&gt;% mutate(labelNam=gsub(&quot;^.*\\.&quot;,&quot;&quot;,gene)) %&gt;% 
  mutate(colourGrp=ifelse(gene %in% genes.to.label$gene, &quot;lab&quot;, 
                          ifelse(gene %in% DEGenesMph1and3$gene, &quot;sig&quot;, &quot;notSig&quot;))) %&gt;% dplyr::filter(gene%in%GeneFreq$gene)

p_allID &lt;- ggscatter(avg.allID, x=&quot;Mph1&quot;, y=&quot;Mph3&quot;,
                     color = &quot;colourGrp&quot;,
                     palette = c(&quot;#B2182B&quot;, &quot;#808080&quot;, &quot;#2166AC&quot;),
                     label= &quot;labelNam&quot;,
                     label.select = genes.to.label$labelNam,
                     repel = T,
                     label.rectangle = F,
                     xlab = &quot;Mph1&quot;,
                     ylab = &quot;Mph3&quot;,
                     font.label = c(&quot;10&quot;, &quot;plain&quot;,&quot;black&quot;),
                     size=2,)
p_allID</code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/scatter%20plot%20DEgenesMph1vsMph3-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-mph1" class="section level2">
<h2>GSEA Mph1</h2>
<pre class="r"><code>## GSEA on genes upregulated in Mph1
##adjust table
DEGenesMph1 &lt;- dplyr::filter(DEGenesMph1and3, cluster == &quot;Mph1&quot;)
DEGenesMph1 &lt;- DEGenesMph1 %&gt;% rownames_to_column(., var = &quot;long&quot;) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, long))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, long))

##GSEA Mph1
egoMph1 &lt;- enrichGO(gene = unique(DEGenesMph1$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoMph1 &lt;- setReadable(egoMph1, OrgDb = org.Mm.eg.db)
dotplot(egoMph1, showCategory=30)</code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/GSEA%20Mph1-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-mph3" class="section level2">
<h2>GSEA Mph3</h2>
<pre class="r"><code>## GSEA on genes upregulated in Mph3
##adjust table
DEGenesMph3 &lt;- dplyr::filter(DEGenesMph1and3, cluster == &quot;Mph3&quot;)
DEGenesMph3 &lt;- DEGenesMph3 %&gt;% rownames_to_column(., var = &quot;long&quot;) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, long))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, long))

##GSEA Mph3
egoMph3 &lt;- enrichGO(gene = unique(DEGenesMph3$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoMph3 &lt;- setReadable(egoMph3, OrgDb = org.Mm.eg.db)
dotplot(egoMph3, showCategory=30)</code></pre>
<p><img src="figure/GranulomaMacrophages.Rmd/GSEA%20Mph3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Jun 12 17:48:53 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.6.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.3.0              
 [4] gridExtra_2.3               DOSE_3.30.1                 enrichplot_1.24.0          
 [7] msigdbr_7.5.1               org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.0      multtest_2.60.0             metap_1.10                 
[13] scater_1.32.0               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.3             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.2                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.0        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.0              S4Vectors_0.42.0            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.3.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.1               Seurat_5.1.0               
[43] SeuratObject_5.0.2          sp_2.1-4                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] zlibbioc_1.50.0           tidyselect_1.2.1          bit_4.0.5                
  [7] doParallel_1.0.17         clue_0.3-65               lattice_0.22-6           
 [10] rjson_0.2.21              blob_1.2.4                S4Arrays_1.4.1           
 [13] pbkrtest_0.5.2            parallel_4.4.0            png_0.1-8                
 [16] plotrix_3.8-4             cli_3.6.2                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] BiocNeighbors_1.22.0      uwot_0.2.2                shadowtext_0.1.3         
 [25] curl_5.2.1                mime_0.12                 evaluate_0.23            
 [28] tidytree_0.4.6            leiden_0.4.3.1            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.4             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] splines_4.4.0             ggraph_2.2.1              sctransform_0.4.1        
 [40] ggbeeswarm_0.7.2          DBI_1.2.3                 jquerylib_0.1.4          
 [43] smoother_1.3              withr_3.0.0               git2r_0.33.0             
 [46] corpcor_1.6.10            class_7.3-22              rprojroot_2.0.4          
 [49] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [61] DEoptimR_1.1-3            reticulate_1.37.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.47               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-1           irlba_2.3.5.1            
 [79] fastDummies_1.7.3         gridGraphics_0.5-1        lazyeval_0.2.2           
 [82] yaml_2.3.8                survival_3.7-0            scattermore_1.2          
 [85] crayon_1.5.2              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [88] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [91] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [94] aod_1.3.3                 KEGGREST_1.44.0           Rtsne_0.17               
 [97] shape_1.4.6.1             limma_3.60.2              pkgconfig_2.0.3          
[100] TMB_1.9.11                mathjaxr_1.6-0            EnvStats_2.8.1           
[103] aplot_0.2.2               scatterplot3d_0.3-44      spatstat.sparse_3.0-3    
[106] ape_5.8                   xtable_1.8-4              car_3.1-2                
[109] highr_0.11                plyr_1.8.9                httr_1.4.7               
[112] rbibutils_2.2.16          tools_4.4.0               globals_0.16.3           
[115] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-164             
[118] lambda.r_1.2.4            HDO.db_0.99.1             lme4_1.1-35.3            
[121] digest_0.6.35             numDeriv_2016.8-1.1       Matrix_1.7-0             
[124] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[127] reshape2_1.4.4            yulab.utils_0.1.4         glue_1.7.0               
[130] cachem_1.1.0              polyclip_1.10-6           generics_0.1.3           
[133] Biostrings_2.72.1         mvtnorm_1.2-5             parallelly_1.37.1        
[136] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[139] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.7              
[142] pbapply_1.7-2             spam_2.10-0               gson_0.1.0               
[145] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[148] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[151] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[154] rmarkdown_2.27            scales_1.3.0              future_1.33.2            
[157] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.0-4      
[160] rstudioapi_0.16.0         cluster_2.1.6             mutoss_0.1-13            
[163] spatstat.utils_3.0-4      hms_1.1.3                 fitdistrplus_1.1-11      
[166] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[169] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[172] xts_0.14.0                dotCall64_1.1-1           shinydashboard_0.7.2     
[175] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-1               
[178] xfun_0.44                 e1071_1.7-14              TH.data_1.1-2            
[181] iterators_1.0.14          abind_1.4-5               GOSemSim_2.30.0          
[184] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-7             
[187] Rdpack_2.6                promises_1.3.0            scatterpie_0.2.3         
[190] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-0           
[193] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[196] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[199] boot_1.3-30               beachmat_2.20.0           listenv_0.9.1            
[202] Rcpp_1.0.12               edgeR_4.2.0               workflowr_1.7.1          
[205] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-60.2            
[208] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[211] spatstat.random_3.2-3     R6_2.5.1                  fastmap_1.2.0            
[214] multcomp_1.4-25           fastmatch_1.1-4           rstatix_0.7.2            
[217] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[220] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-12               
[223] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[226] miniUI_0.1.1.1            deldir_2.0-4              htmltools_0.5.8.1        
[229] ggthemes_5.1.0            bit64_4.0.5               spatstat.explore_3.2-7   
[232] lifecycle_1.0.4           blme_1.0-5                nloptr_2.0.3             
[235] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-2        
[238] spatstat.geom_3.2-9       sn_2.1.1                  ggfun_0.1.5              
[241] future.apply_1.11.2       bslib_0.7.0               pillar_1.9.0             
[244] gplots_3.1.3.1            pcaMethods_1.96.0         locfit_1.5-9.9           
[247] jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.6.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.3.0              
 [4] gridExtra_2.3               DOSE_3.30.1                 enrichplot_1.24.0          
 [7] msigdbr_7.5.1               org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.0      multtest_2.60.0             metap_1.10                 
[13] scater_1.32.0               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.3             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.2                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.0        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.0              S4Vectors_0.42.0            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.3.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.1               Seurat_5.1.0               
[43] SeuratObject_5.0.2          sp_2.1-4                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] zlibbioc_1.50.0           tidyselect_1.2.1          bit_4.0.5                
  [7] doParallel_1.0.17         clue_0.3-65               lattice_0.22-6           
 [10] rjson_0.2.21              blob_1.2.4                S4Arrays_1.4.1           
 [13] pbkrtest_0.5.2            parallel_4.4.0            png_0.1-8                
 [16] plotrix_3.8-4             cli_3.6.2                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] BiocNeighbors_1.22.0      uwot_0.2.2                shadowtext_0.1.3         
 [25] curl_5.2.1                mime_0.12                 evaluate_0.23            
 [28] tidytree_0.4.6            leiden_0.4.3.1            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.4             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] splines_4.4.0             ggraph_2.2.1              sctransform_0.4.1        
 [40] ggbeeswarm_0.7.2          DBI_1.2.3                 jquerylib_0.1.4          
 [43] smoother_1.3              withr_3.0.0               git2r_0.33.0             
 [46] corpcor_1.6.10            class_7.3-22              rprojroot_2.0.4          
 [49] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [61] DEoptimR_1.1-3            reticulate_1.37.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.47               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-1           irlba_2.3.5.1            
 [79] fastDummies_1.7.3         gridGraphics_0.5-1        lazyeval_0.2.2           
 [82] yaml_2.3.8                survival_3.7-0            scattermore_1.2          
 [85] crayon_1.5.2              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [88] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [91] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [94] aod_1.3.3                 KEGGREST_1.44.0           Rtsne_0.17               
 [97] shape_1.4.6.1             limma_3.60.2              pkgconfig_2.0.3          
[100] TMB_1.9.11                mathjaxr_1.6-0            EnvStats_2.8.1           
[103] aplot_0.2.2               scatterplot3d_0.3-44      spatstat.sparse_3.0-3    
[106] ape_5.8                   xtable_1.8-4              car_3.1-2                
[109] highr_0.11                plyr_1.8.9                httr_1.4.7               
[112] rbibutils_2.2.16          tools_4.4.0               globals_0.16.3           
[115] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-164             
[118] lambda.r_1.2.4            HDO.db_0.99.1             lme4_1.1-35.3            
[121] digest_0.6.35             numDeriv_2016.8-1.1       Matrix_1.7-0             
[124] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[127] reshape2_1.4.4            yulab.utils_0.1.4         glue_1.7.0               
[130] cachem_1.1.0              polyclip_1.10-6           generics_0.1.3           
[133] Biostrings_2.72.1         mvtnorm_1.2-5             parallelly_1.37.1        
[136] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[139] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.7              
[142] pbapply_1.7-2             spam_2.10-0               gson_0.1.0               
[145] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[148] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[151] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[154] rmarkdown_2.27            scales_1.3.0              future_1.33.2            
[157] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.0-4      
[160] rstudioapi_0.16.0         cluster_2.1.6             mutoss_0.1-13            
[163] spatstat.utils_3.0-4      hms_1.1.3                 fitdistrplus_1.1-11      
[166] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-0         
[169] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[172] xts_0.14.0                dotCall64_1.1-1           shinydashboard_0.7.2     
[175] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-1               
[178] xfun_0.44                 e1071_1.7-14              TH.data_1.1-2            
[181] iterators_1.0.14          abind_1.4-5               GOSemSim_2.30.0          
[184] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-7             
[187] Rdpack_2.6                promises_1.3.0            scatterpie_0.2.3         
[190] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-0           
[193] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[196] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[199] boot_1.3-30               beachmat_2.20.0           listenv_0.9.1            
[202] Rcpp_1.0.12               edgeR_4.2.0               workflowr_1.7.1          
[205] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-60.2            
[208] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[211] spatstat.random_3.2-3     R6_2.5.1                  fastmap_1.2.0            
[214] multcomp_1.4-25           fastmatch_1.1-4           rstatix_0.7.2            
[217] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[220] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-12               
[223] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[226] miniUI_0.1.1.1            deldir_2.0-4              htmltools_0.5.8.1        
[229] ggthemes_5.1.0            bit64_4.0.5               spatstat.explore_3.2-7   
[232] lifecycle_1.0.4           blme_1.0-5                nloptr_2.0.3             
[235] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-2        
[238] spatstat.geom_3.2-9       sn_2.1.1                  ggfun_0.1.5              
[241] future.apply_1.11.2       bslib_0.7.0               pillar_1.9.0             
[244] gplots_3.1.3.1            pcaMethods_1.96.0         locfit_1.5-9.9           
[247] jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
