<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2024-07-17" />

<title>GranFb</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cxcl13Hpb</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GranFb</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2024-07-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cxcl13Hpb/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240311code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240311code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240311)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allFb_naiveANDd4_seurat.rds
</td>
<td style="text-align:left;">
data/merged_allFb_naiveANDd4_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/DEGenesFbGrancond2.txt
</td>
<td style="text-align:left;">
data/DEGenesFbGrancond2.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_GranFb_seurat.rds
</td>
<td style="text-align:left;">
data/merged_GranFb_seurat.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree49708743e7a501b3b2a0d6c97ca872d1488e8064targetblank4970874a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/49708743e7a501b3b2a0d6c97ca872d1488e8064" target="_blank">4970874</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinCxcl13Hpbtree49708743e7a501b3b2a0d6c97ca872d1488e8064targetblank4970874a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/Cxcl13Hpb/tree/49708743e7a501b3b2a0d6c97ca872d1488e8064" target="_blank">4970874</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/dataJuli2021/
    Ignored:    data/dataJuli2022/
    Ignored:    data/dataJune2019/
    Ignored:    data/dataSept2020/

Untracked files:
    Untracked:  VennDiagram.2025-06-04_10-41-23.549207.log
    Untracked:  VennDiagram.2025-06-04_10-41-23.686772.log
    Untracked:  analysis/FibroblastAnalysisDEgenes_wtonly_withoutGlial.Rmd
    Untracked:  venn-4-diagram.png
    Untracked:  venn-4-diagram.png.2025-06-04_10-41-23.292887.log

Unstaged changes:
    Deleted:    analysis/DEGenesD4cond2.txt
    Deleted:    analysis/DEGenesD4cond3.txt
    Deleted:    analysis/DEGenesFbGrancond2.txt
    Deleted:    analysis/DEGenesFbcond2.txt
    Deleted:    analysis/DEGenesFbcond3.txt
    Deleted:    analysis/DEGenesFbcond3_wtonly.txt
    Modified:   analysis/FRCsignatures.Rmd
    Deleted:    analysis/FbDEGenescond3.txt
    Deleted:    analysis/FbmarkerGenesclustername.txt
    Deleted:    analysis/FbmarkerGenesclustername_wtonly.txt
    Modified:   analysis/FibroblastAnalysisDEgenes.Rmd
    Modified:   analysis/FibroblastAnalysisDEgenes_wtonly.Rmd
    Modified:   analysis/FibroblastAnalysis_wtonly.Rmd
    Modified:   analysis/FibroblastAnalysisincl.D8.Rmd
    Deleted:    analysis/GranDEGenescond2.txt
    Modified:   analysis/GranFb.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GranFb.Rmd</code>) and HTML
(<code>docs/GranFb.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/analysis/GranFb.Rmd" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
<td>
may25
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/Cxcl13Hpb/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/GranFb.html" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
<td>
may25
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/4a5925de2fc1d27f0f8e52596c9dd197814d1830/analysis/GranFb.Rmd" target="_blank">4a5925d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-09-18
</td>
<td>
Sept18-24
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/Cxcl13Hpb/4a5925de2fc1d27f0f8e52596c9dd197814d1830/docs/GranFb.html" target="_blank">4a5925d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-09-18
</td>
<td>
Sept18-24
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)</code></pre>
</div>
<div id="load-object-merged-allfb-d4" class="section level2">
<h2>load object merged allFb d4</h2>
<pre class="r"><code>##load object merged allFb naive and d4
fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allFb_naiveANDd4_seurat.rds&quot;
seuratFb &lt;- readRDS(fileNam)
table(seuratFb$dataset)</code></pre>
<pre><code>
                10_20190604_Mouse_LP_Cxcl13cre_IL33flfl_EYFP_v3 
                                                           6591 
                       10_20200923_Mu_LP_Cxcl13Cre_EYFP_HpD4_v3 
                                                           7409 
        10_20210624_Mu_granuloma_Cxcl13Cre_IL33flf_EYFP_HpD4_v3 
                                                            627 
                         11_20190604_Mouse_LP_Cxcl13cre_EYFP_v3 
                                                           4561 
                         12_20190604_Mouse_LP_Cxcl13cre_EYFP_v3 
                                                           7301 
               7_20200923_Mu_LP_Cxcl13Cre_IL33flfl_EYFP_HpD4_v3 
                                                           7816 
               8_20200923_Mu_LP_Cxcl13Cre_IL33fllf_EYFP_HpD4_v3 
                                                           5899 
                 9_20190604_Mouse_LP_Cxcl13cre_IL33flfl_EYFP_v3 
                                                           6792 
                        9_20200923_Mu_LP_Cxcl13Cre_EYFP_HpD4_v3 
                                                           8550 
                 9_20210624_Mu_granuloma_Cxcl13Cre_EYFP_HpD4_v3 
                                                           1177 
       o289411_1-1_20220629_Mu_granuloma_Cxcl13Cre_CD45_HpD4_v3 
                                                              4 
        o289411_2-2_20220629_Mu_granuloma_Cxcl13Cre_FSC_HpD4_v3 
                                                           2319 
o289411_3-3_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_CD45_HpD4_v3 
                                                              7 
 o289411_4-4_20220629_Mu_granuloma_Cxcl13Cre_IL33fl_FSC_HpD4_v3 
                                                           5165 </code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color vectors</h2>
<pre class="r"><code>## set vector for cond2
colcond2 &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;)
names(colcond2) &lt;- c(&quot;WT&quot;, &quot;cko&quot;)

## set vector for cond2
colcond3 &lt;- c(&quot;#B45B5C&quot;,&quot;#628395&quot;,&quot;#BF782D&quot;)
names(colcond3) &lt;- c(&quot;infectedD4&quot;, &quot;naive&quot;, &quot;granulomaD4&quot;)

#colPal &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#779d8d&quot;,&quot;#0F1F38&quot;,&quot;#E84A5F&quot;,&quot;#FF847C&quot;,&quot;#F8B195&quot;, &quot;#727077&quot;, &quot;#C06C84&quot;,&quot;#2A363B&quot;, &quot;#6C5B7B&quot;)
#names(colPal) &lt;- c(&quot;0&quot; ,&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;)

colPal &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#99B898&quot;,&quot;#E84A5F&quot;,&quot;#0F1F38&quot;,&quot;#FF847C&quot;,&quot;#F8B195&quot;,&quot;#6C5B7B&quot;, &quot;#C06C84&quot;, &quot;#727077&quot;)
names(colPal) &lt;- c(&quot;0&quot; ,&quot;1&quot;, &quot;2&quot;, &quot;3&quot;,&quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;)

colclustername &lt;- c(&quot;#355C7D&quot;, &quot;#8E9B97&quot;,&quot;#99B898&quot;,&quot;#E84A5F&quot;,&quot;#0F1F38&quot;,&quot;#F8B195&quot;,&quot;#727077&quot;,&quot;#FF847C&quot;,&quot;#C06C84&quot;,&quot;#904D39&quot;)
names(colclustername) &lt;- c(&quot;PdgfraloFb1&quot; ,&quot;PdgfraloFb2&quot;, &quot;Trophocytes&quot;,&quot;PdgfraloFb3&quot;, &quot;Telocytes&quot;,&quot;Myocytes&quot;, &quot;Pdgfrahi&quot;, &quot;PdgfraloFb4&quot;, &quot;Thy1Fb&quot;, &quot;Glial&quot;)

colcelltype &lt;- c(&quot;#355C7D&quot;,&quot;#99B898&quot;,&quot;#0F1F38&quot;,&quot;#F8B195&quot;,&quot;#727077&quot;,&quot;#C06C84&quot;,&quot;#904D39&quot;)
names(colcelltype) &lt;- c(&quot;PdgfraloFb&quot;, &quot;Trophocytes&quot;, &quot;Telocytes&quot;,&quot;Myocytes&quot;, &quot;Pdgfrahi&quot;, &quot;Thy1Fb&quot;, &quot;Glial&quot;)

colcelltype2 &lt;- c(&quot;#E84A5F&quot;,&quot;#99B898&quot;,&quot;#0F1F38&quot;,&quot;#F8B195&quot;,&quot;#727077&quot;,&quot;#C06C84&quot;,&quot;#E84A5F&quot;,&quot;#99B898&quot;,&quot;#0F1F38&quot;,&quot;#F8B195&quot;,&quot;#727077&quot;,&quot;#C06C84&quot;)
names(colcelltype2) &lt;- c(&quot;PdgfraloFb&quot;, &quot;Trophocytes&quot;, &quot;Telocytes&quot;,&quot;Myocytes&quot;, &quot;Pdgfrahi&quot;, &quot;Thy1Fb&quot;,&quot;PdgfraloFb_out&quot;, &quot;Trophocytes_out&quot;, &quot;Telocytes_out&quot;,&quot;Myocytes_out&quot;, &quot;Pdgfrahi_out&quot;, &quot;Thy1Fb_out&quot;)</code></pre>
</div>
<div id="subset-pdgfralofb3" class="section level2">
<h2>subset PdgfraloFb3</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$cond2
seurat1 &lt;- subset(seuratFb, clustername == &quot;PdgfraloFb3&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
2552 3361 </code></pre>
<pre class="r"><code>DEGenesPdgfraloFb3 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;PdgfraloFb3&quot;)

VlnPlot(object=seurat1, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-1">
Past versions of PdgfraloFb3-1.png
</button>
</p>
<div id="fig-PdgfraloFb3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat1, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-2">
Past versions of PdgfraloFb3-2.png
</button>
</p>
<div id="fig-PdgfraloFb3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat1, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-3">
Past versions of PdgfraloFb3-3.png
</button>
</p>
<div id="fig-PdgfraloFb3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>seurat2 &lt;- subset(seuratFb, clustername == &quot;Telocytes&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
  WT  cko 
1701 3171 </code></pre>
<pre class="r"><code>DEGenesTelocytes &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Telocytes&quot;)

VlnPlot(object=seurat2, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-4">
Past versions of PdgfraloFb3-4.png
</button>
</p>
<div id="fig-PdgfraloFb3-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat2, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-5">
Past versions of PdgfraloFb3-5.png
</button>
</p>
<div id="fig-PdgfraloFb3-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat2, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-6">
Past versions of PdgfraloFb3-6.png
</button>
</p>
<div id="fig-PdgfraloFb3-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>seurat3 &lt;- subset(seuratFb, clustername == &quot;Myocytes&quot;)
table(seurat3$cond2)</code></pre>
<pre><code>
 WT cko 
756 903 </code></pre>
<pre class="r"><code>DEGenesMyocytes &lt;- FindAllMarkers(seurat3, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Myocytes&quot;)

VlnPlot(object=seurat3, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-7">
Past versions of PdgfraloFb3-7.png
</button>
</p>
<div id="fig-PdgfraloFb3-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-7.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat3, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-8">
Past versions of PdgfraloFb3-8.png
</button>
</p>
<div id="fig-PdgfraloFb3-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-8.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seurat3, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2)</code></pre>
<p><img src="figure/GranFb.Rmd/PdgfraloFb3-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PdgfraloFb3-9">
Past versions of PdgfraloFb3-9.png
</button>
</p>
<div id="fig-PdgfraloFb3-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/PdgfraloFb3-9.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="violin-plots-fb" class="section level2">
<h2>violin plots Fb</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$cond2
VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-1">
Past versions of "violin plot Fb-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-2">
Past versions of "violin plot Fb-2.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$clustername
VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-3">
Past versions of "violin plot Fb-3.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-4">
Past versions of "violin plot Fb-4.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-5">
Past versions of "violin plot Fb-5.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-6">
Past versions of "violin plot Fb-6.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-7">
Past versions of "violin plot Fb-7.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-7.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-8">
Past versions of "violin plot Fb-8.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-8.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-9">
Past versions of "violin plot Fb-9.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-9.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Fb-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Fb-10">
Past versions of "violin plot Fb-10.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Fb-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Fb-10.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="assign-cell-types-and-gran-in-and-out" class="section level2">
<h2>assign cell types and Gran in and out</h2>
<pre class="r"><code>#### celltype
seuratFb$celltype &lt;- &quot;celltype&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;PdgfraloFb1&quot;, &quot;PdgfraloFb2&quot;, &quot;PdgfraloFb3&quot;,&quot;PdgfraloFb4&quot;))] &lt;- &quot;PdgfraloFb&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Trophocytes&quot;))] &lt;- &quot;Trophocytes&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Telocytes&quot;))] &lt;- &quot;Telocytes&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Myocytes&quot;))] &lt;- &quot;Myocytes&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Pdgfrahi&quot;))] &lt;- &quot;Pdgfrahi&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Thy1Fb&quot;))] &lt;- &quot;Thy1Fb&quot;
seuratFb$celltype[which(seuratFb$clustername %in% c(&quot;Glial&quot;))] &lt;- &quot;Glial&quot;
table(seuratFb$celltype)</code></pre>
<pre><code>
      Glial    Myocytes    Pdgfrahi  PdgfraloFb   Telocytes      Thy1Fb Trophocytes 
        154        1659        1313       42982        4872         210       13028 </code></pre>
<pre class="r"><code>###order
Idents(seuratFb) &lt;- seuratFb$celltype
seuratFb$celltype &lt;- factor(seuratFb$celltype, levels=c(&quot;PdgfraloFb&quot;, &quot;Trophocytes&quot;, &quot;Telocytes&quot;,&quot;Myocytes&quot;, &quot;Pdgfrahi&quot;, &quot;Thy1Fb&quot;, &quot;Glial&quot;))
Idents(seuratFb) &lt;- seuratFb$celltype
table(seuratFb$celltype)</code></pre>
<pre><code>
 PdgfraloFb Trophocytes   Telocytes    Myocytes    Pdgfrahi      Thy1Fb       Glial 
      42982       13028        4872        1659        1313         210         154 </code></pre>
<pre class="r"><code>#### Gran in and out
seuratFb$Gran &lt;- &quot;Gran&quot;
seuratFb$Gran[which(seuratFb$cond3 %in% c(&quot;infectedD4&quot;, &quot;naive&quot;))] &lt;- &quot;out&quot;
table(seuratFb$Gran)</code></pre>
<pre><code>
 Gran   out 
 9299 54919 </code></pre>
</div>
<div id="subset-granuloma-fibroblasts" class="section level2">
<h2>subset Granuloma Fibroblasts</h2>
<pre class="r"><code>seuratFbGran &lt;- subset(seuratFb, cond3 == &quot;granulomaD4&quot;)
table(seuratFbGran$cond3)</code></pre>
<pre><code>
granulomaD4 
       9299 </code></pre>
<pre class="r"><code>table(seuratFbGran$cond3)</code></pre>
<pre><code>
granulomaD4 
       9299 </code></pre>
</div>
<div id="subset-fibroblasts-ouside-granuloma" class="section level2">
<h2>subset Fibroblasts ouside Granuloma</h2>
<pre class="r"><code>seuratFbGranout &lt;- subset(seuratFb, cond3 == &quot;granulomaD4&quot;, invert = TRUE)
table(seuratFb$cond3)</code></pre>
<pre><code>
      naive  infectedD4 granulomaD4 
      25245       29674        9299 </code></pre>
<pre class="r"><code>table(seuratFbGranout$cond3)</code></pre>
<pre><code>
     naive infectedD4 
     25245      29674 </code></pre>
</div>
<div id="subset-infected-d4-fibroblasts" class="section level2">
<h2>subset infected d4 Fibroblasts</h2>
<pre class="r"><code>seuratFbSI &lt;- subset(seuratFb, cond3 == &quot;infectedD4&quot;)
table(seuratFb$cond3)</code></pre>
<pre><code>
      naive  infectedD4 granulomaD4 
      25245       29674        9299 </code></pre>
<pre class="r"><code>table(seuratFbSI$cond3)</code></pre>
<pre><code>
infectedD4 
     29674 </code></pre>
</div>
<div id="subset-naive-fibroblasts" class="section level2">
<h2>subset naive Fibroblasts</h2>
<pre class="r"><code>seuratFbnaive &lt;- subset(seuratFb, cond3 == &quot;naive&quot;)
table(seuratFb$cond3)</code></pre>
<pre><code>
      naive  infectedD4 granulomaD4 
      25245       29674        9299 </code></pre>
<pre class="r"><code>table(seuratFbnaive$cond3)</code></pre>
<pre><code>
naive 
25245 </code></pre>
</div>
<div id="umaps" class="section level2">
<h2>umaps</h2>
<pre class="r"><code>##FbGran
## cluster name
Idents(seuratFbGran) &lt;- seuratFbGran$clustername
DimPlot(seuratFbGran, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGran-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGran-1">
Past versions of "umaps FbGran-1.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGran-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGran-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGran, reduction= &quot;pca&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGran-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGran-2">
Past versions of "umaps FbGran-2.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGran-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGran-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGran, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGran-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGran-3">
Past versions of "umaps FbGran-3.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGran-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGran-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cond2
Idents(seuratFbGran) &lt;- seuratFbGran$cond2
DimPlot(seuratFbGran, reduction= &quot;umap&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGran-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGran-4">
Past versions of "umaps FbGran-4.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGran-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGran-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGran, reduction= &quot;pca&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGran-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGran-5">
Past versions of "umaps FbGran-5.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGran-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGran-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>##FbGran outside
## cluster name
Idents(seuratFbGranout) &lt;- seuratFbGranout$clustername
DimPlot(seuratFbGranout, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGranout-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGranout-1">
Past versions of "umaps FbGranout-1.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGranout-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGranout-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGranout, reduction= &quot;pca&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGranout-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGranout-2">
Past versions of "umaps FbGranout-2.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGranout-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGranout-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGranout, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGranout-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGranout-3">
Past versions of "umaps FbGranout-3.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGranout-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGranout-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cond2
Idents(seuratFbGranout) &lt;- seuratFbGranout$cond2
DimPlot(seuratFbGranout, reduction= &quot;umap&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGranout-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGranout-4">
Past versions of "umaps FbGranout-4.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGranout-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGranout-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGranout, reduction= &quot;pca&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbGranout-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbGranout-5">
Past versions of "umaps FbGranout-5.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbGranout-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbGranout-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>##FbSI
## cluster name
Idents(seuratFbSI) &lt;- seuratFbSI$clustername
DimPlot(seuratFbSI, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbSI-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbSI-1">
Past versions of "umaps FbSI-1.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbSI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbSI-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbSI, reduction= &quot;pca&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbSI-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbSI-2">
Past versions of "umaps FbSI-2.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbSI-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbSI-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbSI, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbSI-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbSI-3">
Past versions of "umaps FbSI-3.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbSI-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbSI-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cond2
Idents(seuratFbSI) &lt;- seuratFbSI$cond2
DimPlot(seuratFbSI, reduction= &quot;umap&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbSI-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbSI-4">
Past versions of "umaps FbSI-4.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbSI-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbSI-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbSI, reduction= &quot;pca&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20FbSI-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-FbSI-5">
Past versions of "umaps FbSI-5.png"
</button>
</p>
<div id="fig-no-spaces-umaps-FbSI-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps FbSI-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>##Fbnaive
## cluster name
Idents(seuratFbnaive) &lt;- seuratFbnaive$clustername
DimPlot(seuratFbnaive, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20Fbnaive-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-Fbnaive-1">
Past versions of "umaps Fbnaive-1.png"
</button>
</p>
<div id="fig-no-spaces-umaps-Fbnaive-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps Fbnaive-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbnaive, reduction= &quot;pca&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20Fbnaive-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-Fbnaive-2">
Past versions of "umaps Fbnaive-2.png"
</button>
</p>
<div id="fig-no-spaces-umaps-Fbnaive-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps Fbnaive-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbnaive, reduction= &quot;umap&quot;, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20Fbnaive-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-Fbnaive-3">
Past versions of "umaps Fbnaive-3.png"
</button>
</p>
<div id="fig-no-spaces-umaps-Fbnaive-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps Fbnaive-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cond2
Idents(seuratFbnaive) &lt;- seuratFbnaive$cond2
DimPlot(seuratFbnaive, reduction= &quot;umap&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20Fbnaive-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-Fbnaive-4">
Past versions of "umaps Fbnaive-4.png"
</button>
</p>
<div id="fig-no-spaces-umaps-Fbnaive-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps Fbnaive-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbnaive, reduction= &quot;pca&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/umaps%20Fbnaive-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umaps-Fbnaive-5">
Past versions of "umaps Fbnaive-5.png"
</button>
</p>
<div id="fig-no-spaces-umaps-Fbnaive-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/umaps Fbnaive-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="calculate-de-genes-according-to-cond2" class="section level2">
<h2>calculate DE genes according to cond2</h2>
<pre class="r"><code>## calculate DE genes
Idents(seuratFbGran) &lt;- seuratFbGran$cond2
levels(seuratFbGran)

DEGenesFbGrancond2 &lt;- FindAllMarkers(seuratFbGran, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) 

write.table(DEGenesFbGrancond2, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/DEGenesFbGrancond2.txt&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="de-genes-for-each-celltype-according-to-cond2"
class="section level2">
<h2>DE genes for each celltype according to cond2</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$cond2
#top 100 DE genes PdgfraloFb
seurat1 &lt;- subset(seuratFb, celltype == &quot;PdgfraloFb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
   WT   cko 
21585 21397 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;PdgfraloFb&quot;)

DEGenesPdgfraloFb_100 &lt;- DEGenesPdgfraloFb %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Trophocytes
seurat1 &lt;- subset(seuratFb, celltype == &quot;Trophocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
6582 6446 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTrophocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Trophocytes&quot;)

DEGenesTrophocytes_100 &lt;- DEGenesTrophocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Telocytes
seurat1 &lt;- subset(seuratFb, celltype == &quot;Telocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
1701 3171 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTelocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Telocytes&quot;)

DEGenesTelocytes_100 &lt;- DEGenesTelocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Myocytes
seurat1 &lt;- subset(seuratFb, celltype == &quot;Myocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
756 903 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesMyocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Myocytes&quot;)

DEGenesMyocytes_100 &lt;- DEGenesMyocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Pdgfrahi
seurat1 &lt;- subset(seuratFb, celltype == &quot;Pdgfrahi&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
523 790 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfrahi &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Pdgfrahi&quot;)

DEGenesPdgfrahi_100 &lt;- DEGenesPdgfrahi %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Thy1Fb
seurat1 &lt;- subset(seuratFb, celltype == &quot;Thy1Fb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
111  99 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesThy1Fb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Thy1Fb&quot;)

DEGenesThy1Fb_100 &lt;- DEGenesThy1Fb %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Glial
seurat1 &lt;- subset(seuratFb, celltype == &quot;Glial&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
 63  91 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesGlial &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Glial&quot;)

DEGenesGlial_100 &lt;- DEGenesGlial %&gt;% top_n(100, avg_log2FC) 

DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb_100, DEGenesTrophocytes_100, DEGenesTelocytes_100, DEGenesMyocytes_100, DEGenesPdgfrahi_100, DEGenesThy1Fb_100, DEGenesGlial_100)</code></pre>
</div>
<div
id="distribution-of-logfc-of-overall-top-100-celltype-wise-de-genes"
class="section level2">
<h2>distribution of logFC of overall top 100 celltype wise DE genes</h2>
<pre class="r"><code>ggdensity(DEGenesAll100, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;celltype&quot;, fill = &quot;celltype&quot;, palette = colcelltype)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-1">
Past versions of "boxplot top100 DE genes celltype-1.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;celltype&quot;, palette = colcelltype, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-2">
Past versions of "boxplot top100 DE genes celltype-2.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, color = &quot;celltype&quot;, palette = colcelltype ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-3">
Past versions of "boxplot top100 DE genes celltype-3.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="de-genes-for-each-celltype-according-to-cond2-fbgran-only"
class="section level2">
<h2>DE genes for each celltype according to cond2 !! FbGran only !!</h2>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$cond2
table(seuratFbGran$cond2)</code></pre>
<pre><code>
  WT  cko 
3500 5799 </code></pre>
<pre class="r"><code>table(seuratFbGran$celltype)</code></pre>
<pre><code>
 PdgfraloFb Trophocytes   Telocytes    Myocytes    Pdgfrahi      Thy1Fb       Glial 
       5232         558        2038         649         518         150         154 </code></pre>
<pre class="r"><code>#top 100 DE genes PdgfraloFb
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;PdgfraloFb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
1837 3395 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;PdgfraloFb&quot;)

DEGenesPdgfraloFb_100 &lt;- DEGenesPdgfraloFb %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfraloFb_50 &lt;- DEGenesPdgfraloFb %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Trophocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Trophocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
317 241 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTrophocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Trophocytes&quot;)

DEGenesTrophocytes_100 &lt;- DEGenesTrophocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesTrophocytes_50 &lt;- DEGenesTrophocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Telocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Telocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
 653 1385 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTelocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Telocytes&quot;)

DEGenesTelocytes_100 &lt;- DEGenesTelocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesTelocytes_50 &lt;- DEGenesTelocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Myocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Myocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
332 317 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesMyocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Myocytes&quot;)

DEGenesMyocytes_100 &lt;- DEGenesMyocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesMyocytes_50 &lt;- DEGenesMyocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Pdgfrahi
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Pdgfrahi&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
221 297 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfrahi &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Pdgfrahi&quot;)

DEGenesPdgfrahi_100 &lt;- DEGenesPdgfrahi %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfrahi_50 &lt;- DEGenesPdgfrahi %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Thy1Fb
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Thy1Fb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
 77  73 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesThy1Fb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Thy1Fb&quot;)

DEGenesThy1Fb_100 &lt;- DEGenesThy1Fb %&gt;% top_n(100, avg_log2FC) 
DEGenesThy1Fb_50 &lt;- DEGenesThy1Fb %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Glial
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Glial&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
 63  91 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesGlial &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Glial&quot;)

DEGenesGlial_100 &lt;- DEGenesGlial %&gt;% top_n(100, avg_log2FC) 
DEGenesGlial_50 &lt;- DEGenesGlial %&gt;% top_n(50, avg_log2FC) 

DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb_100, DEGenesTrophocytes_100, DEGenesTelocytes_100, DEGenesMyocytes_100, DEGenesPdgfrahi_100, DEGenesThy1Fb_100, DEGenesGlial_100)

DEGenesAll50 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb_50, DEGenesTrophocytes_50, DEGenesTelocytes_50, DEGenesMyocytes_50, DEGenesPdgfrahi_50, DEGenesThy1Fb_50, DEGenesGlial_50)</code></pre>
</div>
<div
id="distribution-of-logfc-of-overall-top-100-celltype-wise-de-genes-granfb-only"
class="section level2">
<h2>distribution of logFC of overall top 100 celltype wise DE genes !!
GranFb only!!</h2>
<pre class="r"><code>ggdensity(DEGenesAll100, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;celltype&quot;, fill = &quot;celltype&quot;, palette = colcelltype)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-1">
Past versions of "boxplot top100 DE genes celltype GranFb only-1.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggdensity(DEGenesAll50, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;celltype&quot;, fill = &quot;celltype&quot;, palette = colcelltype)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-2">
Past versions of "boxplot top100 DE genes celltype GranFb only-2.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;celltype&quot;, palette = colcelltype, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-3">
Past versions of "boxplot top100 DE genes celltype GranFb only-3.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll50, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;celltype&quot;, palette = colcelltype, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-4">
Past versions of "boxplot top100 DE genes celltype GranFb only-4.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, color = &quot;celltype&quot;, palette = colcelltype ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-5">
Past versions of "boxplot top100 DE genes celltype GranFb only-5.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll50, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, color = &quot;celltype&quot;, palette = colcelltype ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20only-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-6">
Past versions of "boxplot top100 DE genes celltype GranFb only-6.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-only-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb only-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="de-genes-for-each-celltype-according-to-cond2-fbgran-vs-fboutside"
class="section level2">
<h2>DE genes for each celltype according to cond2 !! FbGran vs Fboutside
!!</h2>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$cond2
Idents(seuratFbGranout) &lt;- seuratFbGranout$cond2
table(seuratFbGran$cond2)</code></pre>
<pre><code>
  WT  cko 
3500 5799 </code></pre>
<pre class="r"><code>table(seuratFbGranout$cond2)</code></pre>
<pre><code>
   WT   cko 
27821 27098 </code></pre>
<pre class="r"><code>table(seuratFbGran$celltype)</code></pre>
<pre><code>
 PdgfraloFb Trophocytes   Telocytes    Myocytes    Pdgfrahi      Thy1Fb       Glial 
       5232         558        2038         649         518         150         154 </code></pre>
<pre class="r"><code>table(seuratFbGranout$celltype)</code></pre>
<pre><code>
 PdgfraloFb Trophocytes   Telocytes    Myocytes    Pdgfrahi      Thy1Fb 
      37750       12470        2834        1010         795          60 </code></pre>
<pre class="r"><code>#top 100 DE genes PdgfraloFb Gran
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;PdgfraloFb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
1837 3395 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;PdgfraloFb&quot;)

DEGenesPdgfraloFb_100 &lt;- DEGenesPdgfraloFb %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfraloFb_50 &lt;- DEGenesPdgfraloFb %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes PdgfraloFb outside
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;PdgfraloFb&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
   WT   cko 
19748 18002 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFbout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;PdgfraloFb_out&quot;)

DEGenesPdgfraloFbout_100 &lt;- DEGenesPdgfraloFbout %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfraloFbout_50 &lt;- DEGenesPdgfraloFbout %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Trophocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Trophocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
317 241 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTrophocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Trophocytes&quot;)

DEGenesTrophocytes_100 &lt;- DEGenesTrophocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesTrophocytes_50 &lt;- DEGenesTrophocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Trophocytes outside
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;Trophocytes&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
  WT  cko 
6265 6205 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTrophocytesout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Trophocytes_out&quot;)

DEGenesTrophocytesout_100 &lt;- DEGenesTrophocytesout %&gt;% top_n(100, avg_log2FC) 
DEGenesTrophocytesout_50 &lt;- DEGenesTrophocytesout %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Telocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Telocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
 653 1385 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTelocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Telocytes&quot;)

DEGenesTelocytes_100 &lt;- DEGenesTelocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesTelocytes_50 &lt;- DEGenesTelocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Telocytes outside
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;Telocytes&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
  WT  cko 
1048 1786 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTelocytesout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Telocytes_out&quot;)

DEGenesTelocytesout_100 &lt;- DEGenesTelocytesout %&gt;% top_n(100, avg_log2FC) 
DEGenesTelocytesout_50 &lt;- DEGenesTelocytesout %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Myocytes
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Myocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
332 317 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesMyocytes &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Myocytes&quot;)

DEGenesMyocytes_100 &lt;- DEGenesMyocytes %&gt;% top_n(100, avg_log2FC) 
DEGenesMyocytes_50 &lt;- DEGenesMyocytes %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Myocytes out
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;Myocytes&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
 WT cko 
424 586 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesMyocytesout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Myocytes_out&quot;)

DEGenesMyocytesout_100 &lt;- DEGenesMyocytesout %&gt;% top_n(100, avg_log2FC) 
DEGenesMyocytesout_50 &lt;- DEGenesMyocytesout %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Pdgfrahi
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Pdgfrahi&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
221 297 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfrahi &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Pdgfrahi&quot;)

DEGenesPdgfrahi_100 &lt;- DEGenesPdgfrahi %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfrahi_50 &lt;- DEGenesPdgfrahi %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Pdgfrahi out
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;Pdgfrahi&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
 WT cko 
302 493 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfrahiout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Pdgfrahi_out&quot;)

DEGenesPdgfrahiout_100 &lt;- DEGenesPdgfrahiout %&gt;% top_n(100, avg_log2FC) 
DEGenesPdgfrahiout_50 &lt;- DEGenesPdgfrahiout %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Thy1Fb
seurat1 &lt;- subset(seuratFbGran, celltype == &quot;Thy1Fb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
 77  73 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesThy1Fb &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Thy1Fb&quot;)

DEGenesThy1Fb_100 &lt;- DEGenesThy1Fb %&gt;% top_n(100, avg_log2FC) 
DEGenesThy1Fb_50 &lt;- DEGenesThy1Fb %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes Thy1Fb out
seurat2 &lt;- subset(seuratFbGranout, celltype == &quot;Thy1Fb&quot;)
table(seurat2$cond2)</code></pre>
<pre><code>
 WT cko 
 34  26 </code></pre>
<pre class="r"><code>levels(seurat2) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesThy1Fbout &lt;- FindAllMarkers(seurat2, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(celltype = &quot;Thy1Fb_out&quot;)

DEGenesThy1Fbout_100 &lt;- DEGenesThy1Fbout %&gt;% top_n(100, avg_log2FC) 
DEGenesThy1Fbout_50 &lt;- DEGenesThy1Fbout %&gt;% top_n(50, avg_log2FC) 



DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb_100, DEGenesPdgfraloFbout_100,
                                  DEGenesTrophocytes_100,DEGenesTrophocytesout_100,
                                  DEGenesTelocytes_100,DEGenesTelocytesout_100,
                                  DEGenesMyocytes_100, DEGenesMyocytesout_100,
                                  DEGenesPdgfrahi_100, DEGenesPdgfrahiout_100, 
                                  DEGenesThy1Fb_100, DEGenesThy1Fbout_100)

DEGenesAll50 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb_50,DEGenesPdgfraloFbout_50,
                                 DEGenesTrophocytes_50, DEGenesTrophocytesout_50,
                                 DEGenesTelocytes_50,DEGenesTelocytesout_50,
                                 DEGenesMyocytes_50, DEGenesMyocytesout_50,
                                 DEGenesPdgfrahi_50, DEGenesPdgfrahiout_50,
                                 DEGenesThy1Fb_50, DEGenesThy1Fbout_50)</code></pre>
</div>
<div
id="distribution-of-logfc-of-overall-top-100-celltype-wise-de-genes-granfb-vs-fboutside"
class="section level2">
<h2>distribution of logFC of overall top 100 celltype wise DE genes !!
GranFb vs Fboutside !!</h2>
<pre class="r"><code>ggdensity(DEGenesAll100, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;celltype&quot;, fill = &quot;celltype&quot;, palette = colcelltype2)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-1">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-1.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggdensity(DEGenesAll50, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;celltype&quot;, fill = &quot;celltype&quot;, palette = colcelltype2)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-2">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-2.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;celltype&quot;, palette = colcelltype2, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-3">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-3.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll50, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;celltype&quot;, palette = colcelltype2, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-4">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-4.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll100, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, color = &quot;celltype&quot;, palette = colcelltype2 ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-5">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-5.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll50, x = &quot;celltype&quot;, y = &quot;avg_log2FC&quot;, color = &quot;celltype&quot;, palette = colcelltype2 ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20celltype%20GranFb%20vs%20Fb%20oustide-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-6">
Past versions of "boxplot top100 DE genes celltype GranFb vs Fb
oustide-6.png"
</button>
</p>
<div
id="fig-no-spaces-boxplot-top100-DE-genes-celltype-GranFb-vs-Fb-oustide-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes celltype GranFb vs Fb oustide-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="de-genes-for-each-cluster-according-to-cond2"
class="section level2">
<h2>DE genes for each cluster according to cond2</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$cond2
#top 100 DE genes PdgfraloFb1
seurat1 &lt;- subset(seuratFb, clustername == &quot;PdgfraloFb1&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
   WT   cko 
11352 10566 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb1 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;PdgfraloFb1&quot;)

DEGenesPdgfraloFb1_100 &lt;- DEGenesPdgfraloFb1 %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes PdgfraloFb2
seurat1 &lt;- subset(seuratFb, clustername == &quot;PdgfraloFb2&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
7278 6728 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb2 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;PdgfraloFb2&quot;)

DEGenesPdgfraloFb2_100 &lt;- DEGenesPdgfraloFb2 %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes PdgfraloFb3
seurat1 &lt;- subset(seuratFb, clustername == &quot;PdgfraloFb3&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
2552 3361 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb3 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;PdgfraloFb3&quot;)

DEGenesPdgfraloFb3_100 &lt;- DEGenesPdgfraloFb3 %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes PdgfraloFb4
seurat1 &lt;- subset(seuratFb, clustername == &quot;PdgfraloFb4&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
403 742 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfraloFb4 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;PdgfraloFb4&quot;)

DEGenesPdgfraloFb4_100 &lt;- DEGenesPdgfraloFb4 %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Trophocytes
seurat1 &lt;- subset(seuratFb, clustername == &quot;Trophocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
6582 6446 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTrophocytes&lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Trophocytes&quot;)

DEGenesTrophocytes_100 &lt;- DEGenesTrophocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Telocytes
seurat1 &lt;- subset(seuratFb, clustername == &quot;Telocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
1701 3171 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesTelocytes&lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Telocytes&quot;)

DEGenesTelocytes_100 &lt;- DEGenesTelocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Myocytes
seurat1 &lt;- subset(seuratFb, clustername == &quot;Myocytes&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
756 903 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesMyocytes&lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Myocytes&quot;)

DEGenesMyocytes_100 &lt;- DEGenesMyocytes %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Pdgfrahi
seurat1 &lt;- subset(seuratFb, clustername == &quot;Pdgfrahi&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
523 790 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesPdgfrahi&lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Pdgfrahi&quot;)

DEGenesPdgfrahi_100 &lt;- DEGenesPdgfrahi %&gt;% top_n(100, avg_log2FC) 


DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb1_100, DEGenesPdgfraloFb2_100, DEGenesPdgfraloFb3_100,DEGenesPdgfraloFb4_100, DEGenesTrophocytes_100, DEGenesTelocytes_100, DEGenesMyocytes_100, DEGenesPdgfrahi_100)

#top 100 DE genes Thy1Fb
seurat1 &lt;- subset(seuratFb, clustername == &quot;Thy1Fb&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
111  99 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesThy1Fb&lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Thy1Fb&quot;)

DEGenesThy1Fb_100 &lt;- DEGenesThy1Fb %&gt;% top_n(100, avg_log2FC) 

#top 100 DE genes Glial
seurat1 &lt;- subset(seuratFb, clustername == &quot;Glial&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
 WT cko 
 63  91 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesGlial &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(clustername = &quot;Glial&quot;)

DEGenesGlial_100 &lt;- DEGenesGlial %&gt;% top_n(100, avg_log2FC) 

DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesPdgfraloFb1_100, DEGenesPdgfraloFb2_100, DEGenesPdgfraloFb3_100,DEGenesPdgfraloFb4_100, DEGenesTrophocytes_100, DEGenesTelocytes_100, DEGenesMyocytes_100, DEGenesPdgfrahi_100, DEGenesThy1Fb_100, DEGenesGlial_100)</code></pre>
</div>
<div id="distribution-of-logfc-of-overall-top-100-cw-de-genes"
class="section level2">
<h2>distribution of logFC of overall top 100 cw DE genes</h2>
<pre class="r"><code>ggdensity(DEGenesAll100, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;clustername&quot;, fill = &quot;clustername&quot;, palette = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-cluster-1">
Past versions of "boxplot top100 DE genes cluster-1.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-cluster-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes cluster-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll100, x = &quot;clustername&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;clustername&quot;, palette = colclustername, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20cluster-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-cluster-2">
Past versions of "boxplot top100 DE genes cluster-2.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-cluster-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes cluster-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll100, x = &quot;clustername&quot;, y = &quot;avg_log2FC&quot;, color = &quot;clustername&quot;, palette = colclustername ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20cluster-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-cluster-3">
Past versions of "boxplot top100 DE genes cluster-3.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-cluster-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes cluster-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="de-genes-for-each-cond3-according-to-cond2"
class="section level2">
<h2>DE genes for each cond3 according to cond2</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$cond2
#top 100 DE genes naive
seurat1 &lt;- subset(seuratFb, cond3 == &quot;naive&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
   WT   cko 
11862 13383 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesnaive &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(cond3 = &quot;naive&quot;)

DEGenesnaive_100 &lt;- DEGenesnaive %&gt;% top_n(100, avg_log2FC) 
DEGenesnaive_50 &lt;- DEGenesnaive %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes infectedD4
seurat1 &lt;- subset(seuratFb, cond3 == &quot;infectedD4&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
   WT   cko 
15959 13715 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesinfectedD4 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(cond3 = &quot;infectedD4&quot;)

DEGenesinfectedD4_100 &lt;- DEGenesinfectedD4 %&gt;% top_n(100, avg_log2FC) 
DEGenesinfectedD4_50 &lt;- DEGenesinfectedD4 %&gt;% top_n(50, avg_log2FC) 

#top 100 DE genes granulomaD4
seurat1 &lt;- subset(seuratFb, cond3 == &quot;granulomaD4&quot;)
table(seurat1$cond2)</code></pre>
<pre><code>
  WT  cko 
3500 5799 </code></pre>
<pre class="r"><code>levels(seurat1) </code></pre>
<pre><code>[1] &quot;WT&quot;  &quot;cko&quot;</code></pre>
<pre class="r"><code>DEGenesgranulomaD4 &lt;- FindAllMarkers(seurat1, logfc.threshold = 0.1, only.pos = T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) %&gt;% mutate(cond3 = &quot;granulomaD4&quot;)

DEGenesgranulomaD4_100 &lt;- DEGenesgranulomaD4 %&gt;% top_n(100, avg_log2FC) 
DEGenesgranulomaD4_50 &lt;- DEGenesgranulomaD4 %&gt;% top_n(50, avg_log2FC) 


DEGenesAll100 &lt;- dplyr::bind_rows(DEGenesnaive_100, DEGenesinfectedD4_100, DEGenesgranulomaD4_100)
DEGenesAll50 &lt;- dplyr::bind_rows(DEGenesnaive_50, DEGenesinfectedD4_50, DEGenesgranulomaD4_50)</code></pre>
</div>
<div id="distribution-of-logfc-of-overall-top-100-cond3-wise-de-genes"
class="section level2">
<h2>distribution of logFC of overall top 100 cond3 wise DE genes</h2>
<pre class="r"><code>ggdensity(DEGenesAll100, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;cond3&quot;, fill = &quot;cond3&quot;, palette = colcond3)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-1">
Past versions of "boxplot top100 DE genes overall-1.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggdensity(DEGenesAll50, x = &quot;avg_log2FC&quot;, add= &quot;median&quot;, rug = TRUE, color = &quot;cond3&quot;, fill = &quot;cond3&quot;, palette = colcond3)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-2">
Past versions of "boxplot top100 DE genes overall-2.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll100, x = &quot;cond3&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;cond3&quot;, palette = colcond3, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-3">
Past versions of "boxplot top100 DE genes overall-3.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggviolin(DEGenesAll50, x = &quot;cond3&quot;, y = &quot;avg_log2FC&quot;, fill = &quot;cond3&quot;, palette = colcond3, add = &quot;median_iqr&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-4">
Past versions of "boxplot top100 DE genes overall-4.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll100, x = &quot;cond3&quot;, y = &quot;avg_log2FC&quot;, color = &quot;cond3&quot;, palette = colcond3 ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-5">
Past versions of "boxplot top100 DE genes overall-5.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggboxplot(DEGenesAll50, x = &quot;cond3&quot;, y = &quot;avg_log2FC&quot;, color = &quot;cond3&quot;, palette = colcond3 ,add = &quot;jitter&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/boxplot%20top100%20DE%20genes%20overall-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-top100-DE-genes-overall-6">
Past versions of "boxplot top100 DE genes overall-6.png"
</button>
</p>
<div id="fig-no-spaces-boxplot-top100-DE-genes-overall-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/boxplot top100 DE genes overall-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gsea-fb-gran-wt" class="section level2">
<h2>GSEA Fb gran wt</h2>
<pre class="r"><code>## GSEA on genes upregulated in wt granulomaFb 
##adjust table
DEGenesgranwt &lt;- dplyr::filter(DEGenesgranulomaD4, cluster == &quot;WT&quot;)
DEGenesgranwt &lt;- DEGenesgranwt %&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA Gran
egoGranWT &lt;- enrichGO(gene = unique(DEGenesgranwt$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoGranWT &lt;- setReadable(egoGranWT, OrgDb = org.Mm.eg.db)
dotplot(egoGranWT, showCategory=30)</code></pre>
<p><img src="figure/GranFb.Rmd/GSEA%20Fb%20gran%20wt-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-Fb-gran-wt-1">
Past versions of "GSEA Fb gran wt-1.png"
</button>
</p>
<div id="fig-no-spaces-GSEA-Fb-gran-wt-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/GSEA Fb gran wt-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="violin-plots-infectedd4" class="section level2">
<h2>violin plots infectedD4</h2>
<pre class="r"><code>Idents(seuratFbSI) &lt;- seuratFbSI$cond2
VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-1">
Past versions of "violin plot infectedD4-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-2">
Past versions of "violin plot infectedD4-2.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratFbSI) &lt;- seuratFbSI$clustername
VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-3">
Past versions of "violin plot infectedD4-3.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-4">
Past versions of "violin plot infectedD4-4.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-5">
Past versions of "violin plot infectedD4-5.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbSI, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20infectedD4-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-infectedD4-6">
Past versions of "violin plot infectedD4-6.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-infectedD4-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot infectedD4-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="violin-plots-granfb" class="section level2">
<h2>violin plots GranFb</h2>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$cond2
VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-1">
Past versions of "violin plot GranFb-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-2">
Past versions of "violin plot GranFb-2.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$clustername
VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-3">
Past versions of "violin plot GranFb-3.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-4">
Past versions of "violin plot GranFb-4.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-5">
Past versions of "violin plot GranFb-5.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-6">
Past versions of "violin plot GranFb-6.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-7">
Past versions of "violin plot GranFb-7.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-7.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-8">
Past versions of "violin plot GranFb-8.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-8.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-9">
Past versions of "violin plot GranFb-9.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-9.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-10">
Past versions of "violin plot GranFb-10.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-10.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$mouse
VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-11">
Past versions of "violin plot GranFb-11.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-11.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-12">
Past versions of "violin plot GranFb-12.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-12.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-13">
Past versions of "violin plot GranFb-13.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb-13.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="violin-plots-granfb-wt-only" class="section level2">
<h2>violin plots GranFb wt only</h2>
<pre class="r"><code>seuratFbGranwt &lt;- subset(seuratFbGran, cond2 == &quot;WT&quot;)
## exclude Gial cells
seuratFbGranwt &lt;- subset(seuratFbGranwt, clustername == &quot;Glial&quot;, invert =TRUE)
table(seuratFbGran$cond2)</code></pre>
<pre><code>
  WT  cko 
3500 5799 </code></pre>
<pre class="r"><code>table(seuratFbGranwt$cond2)</code></pre>
<pre><code>
  WT 
3437 </code></pre>
<pre class="r"><code>table(seuratFbGranwt$clustername)</code></pre>
<pre><code>
PdgfraloFb1 PdgfraloFb2 PdgfraloFb3 PdgfraloFb4 Trophocytes   Telocytes    Myocytes    Pdgfrahi 
         53         112        1529         143         317         653         332         221 
     Thy1Fb 
         77 </code></pre>
<pre class="r"><code>Idents(seuratFbGranwt) &lt;- seuratFbGranwt$clustername
VlnPlot(object=seuratFbGranwt, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-1">
Past versions of "violin plot GranFb wt only-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGranwt, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-2">
Past versions of "violin plot GranFb wt only-2.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGranwt, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-3">
Past versions of "violin plot GranFb wt only-3.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGranwt, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-4">
Past versions of "violin plot GranFb wt only-4.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGranwt, features = &quot;ENSMUSG00000026073.Il1r2&quot;, pt.size = 0, cols = colclustername)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-5">
Past versions of "violin plot GranFb wt only-5.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratFbGran) &lt;- seuratFbGran$mouse
VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-6">
Past versions of "violin plot GranFb wt only-6.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-7">
Past versions of "violin plot GranFb wt only-7.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-7.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFbGran, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20GranFb%20wt%20only-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-GranFb-wt-only-8">
Past versions of "violin plot GranFb wt only-8.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-GranFb-wt-only-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot GranFb wt only-8.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="dotplot-fbgranwt" class="section level3">
<h3>dotplot FbGranwt</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratFbGranwt)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=rev(c(&quot;Ccl11&quot;,&quot;Il1r1&quot;,&quot;Osmr&quot;,&quot;Il1r2&quot;))) %&gt;% left_join(., genes, by=&quot;geneID&quot;)

DotPlot(seuratFbGranwt, features = selGenes, group.by= &quot;clustername&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;E&quot;) + coord_flip()</code></pre>
<p><img src="figure/GranFb.Rmd/dotplot%20FbGranwt%20Ccl11%20Osmr%20and%20Il1r1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-FbGranwt-Ccl11-Osmr-and-Il1r1-1">
Past versions of "dotplot FbGranwt Ccl11 Osmr and Il1r1-1.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-FbGranwt-Ccl11-Osmr-and-Il1r1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/dotplot FbGranwt Ccl11 Osmr and Il1r1-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="violin-plots-according-to-gran-in-and-out"
class="section level2">
<h2>violin plots according to Gran in and out</h2>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$clustername
VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000024810.Il33&quot;, pt.size = 0, cols = colclustername, split.by = &quot;Gran&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Gran%20in%20and%20out-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Gran-in-and-out-1">
Past versions of "violin plot Gran in and out-1.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Gran-in-and-out-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Gran in and out-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000020676.Ccl11&quot;, pt.size = 0, cols = colclustername, split.by = &quot;Gran&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Gran%20in%20and%20out-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Gran-in-and-out-2">
Past versions of "violin plot Gran in and out-2.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Gran-in-and-out-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Gran in and out-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000026072.Il1r1&quot;, pt.size = 0, cols = colclustername, split.by = &quot;Gran&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Gran%20in%20and%20out-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Gran-in-and-out-3">
Past versions of "violin plot Gran in and out-3.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Gran-in-and-out-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Gran in and out-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(object=seuratFb, features = &quot;ENSMUSG00000022146.Osmr&quot;, pt.size = 0, cols = colclustername, split.by = &quot;Gran&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/violin%20plot%20Gran%20in%20and%20out-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-plot-Gran-in-and-out-4">
Past versions of "violin plot Gran in and out-4.png"
</button>
</p>
<div id="fig-no-spaces-violin-plot-Gran-in-and-out-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/violin plot Gran in and out-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#rerun seurat GranFb
seuratFbGran &lt;- NormalizeData (object = seuratFbGran)
seuratFbGran &lt;- FindVariableFeatures(object = seuratFbGran)
seuratFbGran &lt;- ScaleData(object = seuratFbGran, verbose = TRUE)
seuratFbGran &lt;- RunPCA(object=seuratFbGran, npcs = 30, verbose = FALSE)
seuratFbGran &lt;- RunTSNE(object=seuratFbGran, reduction=&quot;pca&quot;, dims = 1:20)
seuratFbGran &lt;- RunUMAP(object=seuratFbGran, reduction=&quot;pca&quot;, dims = 1:20)
seuratFbGran &lt;- FindNeighbors(object = seuratFbGran, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratFbGran &lt;- FindClusters(object = seuratFbGran, resolution = res[i], random.seed = 1234)
}

### save seurat object
saveRDS(seuratFbGran, file=&quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_GranFb_seurat.rds&quot;)</code></pre>
</div>
<div id="umaps-1" class="section level2">
<h2>umaps</h2>
<pre class="r"><code>### load object
saveRDS(seuratFbGran, file=&quot;/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_GranFb_seurat.rds&quot;)

##FbGran
## cluster name
Idents(seuratFbGran) &lt;- seuratFbGran$RNA_snn_res.0.25
DimPlot(seuratFbGran, reduction= &quot;umap&quot;) </code></pre>
<p><img src="figure/GranFb.Rmd/load%20and%20umaps%20FbGran-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-load-and-umaps-FbGran-1">
Past versions of "load and umaps FbGran-1.png"
</button>
</p>
<div id="fig-no-spaces-load-and-umaps-FbGran-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/load and umaps FbGran-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGran, reduction= &quot;pca&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/load%20and%20umaps%20FbGran-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-load-and-umaps-FbGran-2">
Past versions of "load and umaps FbGran-2.png"
</button>
</p>
<div id="fig-no-spaces-load-and-umaps-FbGran-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/load and umaps FbGran-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cond2
Idents(seuratFbGran) &lt;- seuratFbGran$cond2
DimPlot(seuratFbGran, reduction= &quot;umap&quot;, cols = colcond2, shuffle = TRUE) </code></pre>
<p><img src="figure/GranFb.Rmd/load%20and%20umaps%20FbGran-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-load-and-umaps-FbGran-3">
Past versions of "load and umaps FbGran-3.png"
</button>
</p>
<div id="fig-no-spaces-load-and-umaps-FbGran-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/load and umaps FbGran-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratFbGran, reduction= &quot;pca&quot;, cols = colcond2, shuffle = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranFb.Rmd/load%20and%20umaps%20FbGran-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-load-and-umaps-FbGran-4">
Past versions of "load and umaps FbGran-4.png"
</button>
</p>
<div id="fig-no-spaces-load-and-umaps-FbGran-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/load and umaps FbGran-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dotplot-fbgran" class="section level2">
<h2>dotplot FbGran</h2>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratFbGran)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=c(&quot;Thy1&quot;,&quot;Ly6a&quot;,&quot;Acta2&quot;,&quot;Cd81&quot;,&quot;Cd34&quot;,&quot;Pdgfra&quot;,&quot;Pdpn&quot;)) %&gt;% left_join(., genes, by=&quot;geneID&quot;)

DotPlot(seuratFbGran, features = selGenes, group.by= &quot;RNA_snn_res.0.25&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;E&quot;) + coord_flip()</code></pre>
<p><img src="figure/GranFb.Rmd/dotplot%20Fb%20Gran-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-Fb-Gran-1">
Past versions of "dotplot Fb Gran-1.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-Fb-Gran-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/dotplot Fb Gran-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DotPlot(seuratFbGran, features = selGenes, group.by= &quot;RNA_snn_res.0.25&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;F&quot;) + coord_flip()</code></pre>
<p><img src="figure/GranFb.Rmd/dotplot%20Fb%20Gran-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-Fb-Gran-2">
Past versions of "dotplot Fb Gran-2.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-Fb-Gran-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/dotplot Fb Gran-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="feature-plots-fb-maker-genes" class="section level2">
<h2>feature plots fb maker genes</h2>
<pre class="r"><code>## plot feature
FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000028583.Pdpn&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-1">
Past versions of "feature plots fb-1.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000028583.Pdpn&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-2">
Past versions of "feature plots fb-2.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-2.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;Rosa26eyfp.Rosa26eyfp&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-3">
Past versions of "feature plots fb-3.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-3.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;Rosa26eyfp.Rosa26eyfp&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-4">
Past versions of "feature plots fb-4.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-4.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000035783.Acta2&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-5">
Past versions of "feature plots fb-5.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-5.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000035783.Acta2&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-6">
Past versions of "feature plots fb-6.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-6.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000016494.Cd34&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-7">
Past versions of "feature plots fb-7.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-7.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000016494.Cd34&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-8">
Past versions of "feature plots fb-8.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-8.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000029231.Pdgfra&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-9">
Past versions of "feature plots fb-9.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-9.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000029231.Pdgfra&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-10">
Past versions of "feature plots fb-10.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-10.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000023078.Cxcl13&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-11">
Past versions of "feature plots fb-11.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-11.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000023078.Cxcl13&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-12">
Past versions of "feature plots fb-12.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-12.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000071005.Ccl19&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-13">
Past versions of "feature plots fb-13.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-13.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000071005.Ccl19&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-14">
Past versions of "feature plots fb-14.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-14.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000037706.Cd81&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-15">
Past versions of "feature plots fb-15.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-15.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000037706.Cd81&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-16">
Past versions of "feature plots fb-16.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-16.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-17">
Past versions of "feature plots fb-17.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-17.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000024810.Il33&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-18">
Past versions of "feature plots fb-18.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-18.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000023078.Cxcl13&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-19">
Past versions of "feature plots fb-19.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-19.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000023078.Cxcl13&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-20">
Past versions of "feature plots fb-20.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-20.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-21">
Past versions of "feature plots fb-21.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-21.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000020676.Ccl11&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-22">
Past versions of "feature plots fb-22.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-22.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000024011.Pi16&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1, order = TRUE)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-23">
Past versions of "feature plots fb-23.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-23.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(seuratFbGran, features = &quot;ENSMUSG00000024011.Pi16&quot;, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), pt.size = 1) + theme(legend.position = &quot;none&quot;) + ggtitle(NULL)</code></pre>
<p><img src="figure/GranFb.Rmd/feature%20plots%20fb-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-plots-fb-24">
Past versions of "feature plots fb-24.png"
</button>
</p>
<div id="fig-no-spaces-feature-plots-fb-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/feature plots fb-24.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="rel-abundance-fb-subsets" class="section level2">
<h2>Rel Abundance Fb subsets</h2>
<pre class="r"><code>## make count list according to cond3
datList &lt;- NULL
for(con in unique(seuratFbGran$cond2)){
  seuratSub &lt;- subset(seuratFbGran, cond2==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$RNA_snn_res.0.25)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(cond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 30749  5799
[1] 30749  3500</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

##order x
ordX &lt;-  c(&quot;WT&quot;, &quot;cko&quot;) 

## plot abundance
ggbarplot(dat_all, x= &quot;cond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;clustername&quot;, ylab = &quot;frequency&quot;) + scale_x_discrete(limits=ordX)</code></pre>
<p><img src="figure/GranFb.Rmd/Fb%20abundance%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fb-abundance-plot-1">
Past versions of "Fb abundance plot-1.png"
</button>
</p>
<div id="fig-no-spaces-Fb-abundance-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/49708743e7a501b3b2a0d6c97ca872d1488e8064/docs/figure/GranFb.Rmd/Fb abundance plot-1.png" target="_blank">4970874</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/66ec1aa819fcef0990ffe1b8b20b1e3ffb3c9be5/docs/figure/GranFb.Rmd/Fb abundance plot-1.png" target="_blank">66ec1aa</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-11-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/Cxcl13Hpb/blob/4a5925de2fc1d27f0f8e52596c9dd197814d1830/docs/figure/GranFb.Rmd/Fb abundance plot-1.png" target="_blank">4a5925d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-09-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Jun  4 11:52:02 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.1              
 [4] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
 [7] msigdbr_10.0.2              org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.12                 
[13] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.4             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.4                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.5.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.2               Seurat_5.3.0               
[43] SeuratObject_5.1.0          sp_2.2-0                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.2         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] gridGraphics_0.5-1        fastDummies_1.7.5         lazyeval_0.2.2           
 [85] yaml_2.3.10               survival_3.8-3            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [91] progressr_0.15.1          tweenr_2.0.3              later_1.4.2              
 [94] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [97] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[100] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[103] TMB_1.9.17                spatstat.univar_3.1-3     mathjaxr_1.8-0           
[106] EnvStats_3.1.0            aplot_0.2.5               scatterplot3d_0.3-44     
[109] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[112] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[115] rbibutils_2.3             tools_4.4.0               globals_0.18.0           
[118] beeswarm_0.4.0            broom_1.0.8               nlme_3.1-168             
[121] lambda.r_1.2.4            assertthat_0.2.1          lme4_1.1-37              
[124] digest_0.6.37             numDeriv_2016.8-1.1       Matrix_1.7-3             
[127] farver_2.1.2              tzdb_0.5.0                remaCor_0.0.18           
[130] reshape2_1.4.4            yulab.utils_0.2.0         glue_1.8.0               
[133] cachem_1.1.0              polyclip_1.10-7           generics_0.1.4           
[136] Biostrings_2.72.1         mvtnorm_1.3-3             presto_1.0.0             
[139] parallelly_1.44.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.49.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-3      hms_1.1.3                
[172] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.2            scatterpie_0.2.4         
[196] RSQLite_2.3.11            qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.3-3     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-2   
[238] lifecycle_1.0.4           blme_1.0-6                msigdbdf_24.1.1          
[241] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[244] robustbase_0.99-4-1       spatstat.geom_3.3-6       sn_2.1.1                 
[247] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[250] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[253] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.1              
 [4] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
 [7] msigdbr_10.0.2              org.Mm.eg.db_3.19.1         AnnotationDbi_1.66.0       
[10] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.12                 
[13] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[16] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[19] viridisLite_0.4.2           lubridate_1.9.4             forcats_1.0.0              
[22] stringr_1.5.1               purrr_1.0.4                 readr_2.1.5                
[25] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[28] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[31] Biobase_2.64.0              GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[34] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[37] MatrixGenerics_1.16.0       matrixStats_1.5.0           pheatmap_1.0.12            
[40] ggpubr_0.6.0                ggplot2_3.5.2               Seurat_5.3.0               
[43] SeuratObject_5.1.0          sp_2.2-0                    runSeurat3_0.1.0           
[46] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.2         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] gridGraphics_0.5-1        fastDummies_1.7.5         lazyeval_0.2.2           
 [85] yaml_2.3.10               survival_3.8-3            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [91] progressr_0.15.1          tweenr_2.0.3              later_1.4.2              
 [94] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [97] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[100] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[103] TMB_1.9.17                spatstat.univar_3.1-3     mathjaxr_1.8-0           
[106] EnvStats_3.1.0            aplot_0.2.5               scatterplot3d_0.3-44     
[109] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[112] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[115] rbibutils_2.3             tools_4.4.0               globals_0.18.0           
[118] beeswarm_0.4.0            broom_1.0.8               nlme_3.1-168             
[121] lambda.r_1.2.4            assertthat_0.2.1          lme4_1.1-37              
[124] digest_0.6.37             numDeriv_2016.8-1.1       Matrix_1.7-3             
[127] farver_2.1.2              tzdb_0.5.0                remaCor_0.0.18           
[130] reshape2_1.4.4            yulab.utils_0.2.0         glue_1.8.0               
[133] cachem_1.1.0              polyclip_1.10-7           generics_0.1.4           
[136] Biostrings_2.72.1         mvtnorm_1.3-3             presto_1.0.0             
[139] parallelly_1.44.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.49.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-3      hms_1.1.3                
[172] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.2            scatterpie_0.2.4         
[196] RSQLite_2.3.11            qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.3-3     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-2   
[238] lifecycle_1.0.4           blme_1.0-6                msigdbdf_24.1.1          
[241] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[244] robustbase_0.99-4-1       spatstat.geom_3.3-6       sn_2.1.1                 
[247] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[250] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[253] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
