---
title: "CellChatGranuloma"
author: "A.DeMartin"
date: "2024-02-21"
output: 
  html_document:
    keep_md: true
    toc: true
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
options(width = 100)
knitr::opts_chunk$set(warning = FALSE, message = FALSE, dev = c("png", "pdf"))
seed <- 1234
```

## load packages
```{r load packages}
library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)
```

## load merged file granuloma
```{r load merged file granuloma}
##load merged file 
fileNam <- "/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/merged_allgranuloma_wofilter_seurat_wocl21.rds"
seurat <- readRDS(fileNam)
table(seurat$dataset)
table(seurat$RNA_snn_res.0.25)
table(seurat$orig.ident)
```

## set color vectors
```{r set color vectors}
### col clustername
colclustername <- c("#355C7D","#B1746FFF","#202547","#B09C85", "#4e5a4c","#53354A","#2A363B","#8491B4FF","#00A087FF","#DC9989","#84ad83", "#628395","#D33B44","#779d8d","#727077","#868686FF","#F8B195","#FF847C","#725663FF","#904D39","#91D1C2")
names(colclustername) <- c("PdgfraloFb1","Mph1", "CD8Tcell1","Neut", "IgABcell1", "LEC", "Telocytes", "DC", "CD4Tcell/ILC2", "Mph2",  "NKcell", "CD8Tcell2", "Mph3", "Trophocytes", "PdgfrahiLy6anegFb", "IgABcell2", "Myocytes", "PdgfraloFb2", "IgDBcell", "NC", "ICC")

### col cond2
colcond2 <- c("#202547","#BE3144")
names(colcond2) <- c("WT", "cko")

table(seurat$cluster_name)
```

################################## Cellchat interaction analysis ####################################

## load packages cellchat
```{r load packages cellchat}
#devtools::install_github("sqjin/CellChat")
library(CellChat)
library(patchwork)
options(stringsAsFactors = FALSE)
```

## assign clusterLabel
```{r assign clusterLabel ext, eval=FALSE, include=TRUE}
seurat$clusterNameext <- "allCells"
seurat$clusterNameext[which(seurat$cluster_name == "CD8Tcell1")] <- "a_CD8Tcell1"
seurat$clusterNameext[which(seurat$cluster_name == "CD8Tcell2")] <- "b_CD8Tcell2"
seurat$clusterNameext[which(seurat$cluster_name == "NKcell")] <- "c_NKcell"
seurat$clusterNameext[which(seurat$cluster_name == "CD4Tcell/ILC2")] <- "d_CD4Tcell/ILC2"
seurat$clusterNameext[which(seurat$cluster_name == "Neut")] <- "e_Neut"
seurat$clusterNameext[which(seurat$cluster_name == "Mph1")] <- "f_Mph1"
seurat$clusterNameext[which(seurat$cluster_name == "Mph2")] <- "g_Mph2"
seurat$clusterNameext[which(seurat$cluster_name == "Mph3")] <- "h_Mph3"
seurat$clusterNameext[which(seurat$cluster_name == "DC")] <- "i_DC"
seurat$clusterNameext[which(seurat$cluster_name == "IgABcell1")] <- "j_IgABcell1"
seurat$clusterNameext[which(seurat$cluster_name == "IgABcell2")] <- "k_IgABcell2"
seurat$clusterNameext[which(seurat$cluster_name == "IgDBcell")] <- "l_IgDBcell"
seurat$clusterNameext[which(seurat$cluster_name == "PdgfraloFb1")] <- "m_PdgfraloFb1"
seurat$clusterNameext[which(seurat$cluster_name == "PdgfraloFb2")] <- "n_PdgfraloFb2"
seurat$clusterNameext[which(seurat$cluster_name == "Trophocytes")] <- "o_Trophocytes"
seurat$clusterNameext[which(seurat$cluster_name == "Telocytes")] <- "p_Telocytes"
seurat$clusterNameext[which(seurat$cluster_name == "Myocytes")] <- "q_Myocytes"
seurat$clusterNameext[which(seurat$cluster_name == "PdgfrahiLy6anegFb")] <- "r_PdgfrahiLy6anegFb"
seurat$clusterNameext[which(seurat$cluster_name == "NC")] <- "s_NC"
seurat$clusterNameext[which(seurat$cluster_name == "LEC")] <- "t_LEC"
seurat$clusterNameext[which(seurat$cluster_name == "ICC")] <- "u_ICC"
table(seurat$clusterNameext)

#separate cko and wt for further CellChat analysis
seuratcko <- subset(seurat, cond2 == "cko")
seuratwt <- subset(seurat, cond2 == "WT")
```

convert to sce to change rownames
```{r make sce, eval=FALSE, include=TRUE}
sce <- as.SingleCellExperiment(seurat)
scecko <- as.SingleCellExperiment(seuratcko)
scewt <- as.SingleCellExperiment(seuratwt)
#rownames(sce) = gsub("\\..*","",rownames(sce))
rownames(sce) = gsub("^.*\\.","",rownames(sce))
rownames(scecko) = gsub("^.*\\.","",rownames(scecko))
rownames(scewt) = gsub("^.*\\.","",rownames(scewt))
```

create a CellChat object
```{r create CellChat object, eval=FALSE, include=TRUE}
cellchat <- createCellChat(object = sce, group.by = "clusterNameext")
cellchatcko <- createCellChat(object = scecko, group.by = "clusterNameext")
cellchatwt <- createCellChat(object = scewt, group.by = "clusterNameext")
#> Create a CellChat object from a data matrix
#> Set cell identities for the new CellChat object
```

add cell information into metaslot (optional)
```{r setup2, include=FALSE, eval=FALSE, include=TRUE}
#cellchat <- addMeta(cellchat, meta = meta)
#cellchat <- setIdent(cellchat, ident.use = "labels") # set "labels" as default cell identity
#levels(cellchat@idents) # show factor levels of the cell labels
#groupSize <- as.numeric(table(cellchat@idents)) # number of cells in each cell group
```

set the ligand receptor interaction database
```{r set database, eval=FALSE, include=TRUE}
CellChatDB <- CellChatDB.mouse # use CellChatDB.human if running on mouse data
showDatabaseCategory(CellChatDB)
# Show the structure of the database
dplyr::glimpse(CellChatDB$interaction)

# use a subset of CellChatDB for cell-cell communication analysis
#CellChatDB.use <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling

# use all CellChatDB for cell-cell communication analysis
CellChatDB.use <- CellChatDB # simply use the default CellChatDB

# set the used database in the object
cellchat@DB <- CellChatDB.use
cellchatcko@DB <- CellChatDB.use
cellchatwt@DB <- CellChatDB.use
```

processing expression data for cellcell communication analysis
```{r processing expression data, eval=FALSE, include=TRUE}
# subset the expression data of signaling genes for saving computation cost
cellchat <- subsetData(cellchat) # This step is necessary even if using the whole database
future::plan("multisession", workers = 4) # do parallel
options(future.globals.maxSize = 10000 * 1024^2)
#> Warning: [ONE-TIME WARNING] Forked processing ('multicore') is disabled
#> in future (>= 1.13.0) when running R from RStudio, because it is
#> considered unstable. Because of this, plan("multicore") will fall
#> back to plan("sequential"), and plan("multiprocess") will fall back to
#> plan("multisession") - not plan("multicore") as in the past. For more details,
#> how to control forked processing or not, and how to silence this warning in
#> future R sessions, see ?future::supportsMulticore
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
# project gene expression data onto PPI network (optional)
# cellchat <- projectData(cellchat, PPI.human)


##same for cko
cellchatcko <- subsetData(cellchatcko) # This step is necessary even if using the whole database
future::plan("multisession", workers = 4) # do parallel
options(future.globals.maxSize = 10000 * 1024^2)

cellchatcko <- identifyOverExpressedGenes(cellchatcko)
cellchatcko <- identifyOverExpressedInteractions(cellchatcko)

##same for wt
cellchatwt <- subsetData(cellchatwt)
future::plan("multisession", workers = 4) 
options(future.globals.maxSize = 10000 * 1024^2)

cellchatwt <- identifyOverExpressedGenes(cellchatwt)
cellchatwt <- identifyOverExpressedInteractions(cellchatwt)
```

Compute the communication probability and infer cellular communication network
```{r compute communication probability, eval=FALSE, include=TRUE}
cellchat <- computeCommunProb(cellchat, type =  "truncatedMean", trim= 0.1) #set truncated mean to 10% - average gene expression is zero if less than 10% of cell in one group express gene (default is 25%)
# Filter out the cell-cell communication if there are only few number of cells in certain cell groups
# cellchat <- filterCommunication(cellchat, min.cells = 35)

cellchatcko <- computeCommunProb(cellchatcko, type =  "truncatedMean", trim= 0.1) 
cellchatwt <- computeCommunProb(cellchatwt, type =  "truncatedMean", trim= 0.1) 
```

Infer the cell-cell communication at a signaling pathway level
CellChat computes the communication probability on signaling pathway level by summarizing the communication probabilities of all ligands-receptors interactions associated with each signaling pathway.
```{r CommunProbPathwqy, eval=FALSE, include=TRUE}
cellchat <- computeCommunProbPathway(cellchat)
cellchatcko <- computeCommunProbPathway(cellchatcko)
cellchatwt <- computeCommunProbPathway(cellchatwt)
```


```{r aggregateNet, eval=FALSE, include=TRUE}
cellchat <- aggregateNet(cellchat)
cellchatcko <- aggregateNet(cellchatcko)
cellchatwt <- aggregateNet(cellchatwt)
```

Compute the network centrality scores
```{r computeCentrality, eval=FALSE, include=TRUE}
future.seed=TRUE
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
future.seed=TRUE
cellchatcko <- netAnalysis_computeCentrality(cellchatcko, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
future.seed=TRUE
cellchatwt <- netAnalysis_computeCentrality(cellchatwt, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
```

```{r save CellChat object, eval=FALSE, include=TRUE}
saveRDS(cellchat, file = "/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_allext.rds")
saveRDS(cellchatcko, file = "/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_cko.rds")
saveRDS(cellchatwt, file = "/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_wt.rds")
```

#####################################analysis dataset with all#######################################

```{r load cellchat object}
cellchat.all <- readRDS("/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_allext.rds")
```

```{r setcolor vector}
colclusterName <- c("#202547","#628395","#84ad83","#779d8d","#B09C85","#B1746FFF","#DC9989","#D33B44","#8491B4FF","#4e5a4c","#727077","#725663FF","#355C7D","#FF847C","#779d8d","#2A363B","#F8B195","#868686FF","#904D39","#53354A","#91D1C2")
```

Extract the inferred cellular communication network as a data frame
We provide a function subsetCommunication to easily access the inferred cell-cell communications of interest
```{r extract interacitons}
df.net <- subsetCommunication(cellchat.all) #returns a data frame consisting of all the inferred cell-cell communications at the level of ligands/receptors. Set slot.name = "netP" to access the the inferred communications at the level of signaling pathways
#interactions immunecells to stroma only
df.net <- subsetCommunication(cellchat.all) 
#sources.use = c(1:11), targets.use = c(12:20)
#sources.use=c("tPI16RC2","TRC3","ACTA2PRC2","ACTA2PRC3","tPI16RC1","ACTA2PRC1","TRC2","TRC1","PI16RC","FDC"), targets.use = c("naiveCD4","Tcm","CTL","Treg","GCBcells","Tfh","naiveBcells","plasmaCells"))
df.net <- subsetCommunication(cellchat.all, signaling = c("BMP", "TGFb")) #gives the inferred cell-cell communications mediated by signaling WNT and TGFb.
```

Calculate the aggregated cell-cell communication network
We can calculate the aggregated cell-cell communication network by counting the number of links or summarizing the communication probability. USER can also calculate the aggregated network among a subset of cell groups by setting sources.use and targets.use.
```{r signaling role}
# Signaling role analysis on the aggregated cell-cell communication network from all signaling pathways
gg1 <- netAnalysis_signalingRole_scatter(cellchat.all, color.use = colclusterName)
gg1
```

```{r signaling heatmap}
par(mfrow=c(1,1))
netVisual_heatmap(cellchat.all, color.heatmap = "GnBu", color.use = colclusterName, measure = "count")
netVisual_heatmap(cellchat.all, color.heatmap = "GnBu", color.use = colclusterName, measure = "weight")
#> Do heatmap based on a single object
```

```{r contributing signals, fig.height=12, fig.width=12}
#Identify signals contributing most to outgoing or incoming signaling of certain cell groups
#We can also answer the question on which signals contributing most to outgoing or incoming signaling of certain cell groups.
# Signaling role analysis on the aggregated cell-cell communication network from all signaling pathways
ht1 <- netAnalysis_signalingRole_heatmap(cellchat.all, pattern = "outgoing", color.use = colclusterName, height = 17)
ht2 <- netAnalysis_signalingRole_heatmap(cellchat.all, pattern = "incoming", color.use = colclusterName, height = 17)
ht1 + ht2
```

```{r interaction btw immune cells and fibroblats}
cellchat.all <- aggregateNet(cellchat.all)
#We can also visualize the aggregated cell-cell communication network. For example, showing the number of interactions or the total interaction strength (weights) between any two cell groups using circle plot.
groupSize <- as.numeric(table(cellchat.all@idents))
par(mfrow = c(1,2), xpd=TRUE)
netVisual_circle(cellchat.all@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Number of interactions", targets.use = 13:18, sources.use = 1:12, color.use = colclusterName)
netVisual_circle(cellchat.all@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Interaction weights/strength", targets.use = 13:18, sources.use = 1:12, color.use = colclusterName)
```

```{r interaction btw Tcells and fibroblats}
groupSize <- as.numeric(table(cellchat.all@idents))
par(mfrow = c(1,2), xpd=TRUE)
netVisual_circle(cellchat.all@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Number of interactions", targets.use = 13:21, sources.use = 1:4, color.use = colclusterName)
netVisual_circle(cellchat.all@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Interaction weights/strength", targets.use = 13:21, sources.use = 1:4, color.use = colclusterName)
```


```{r interaction btw Macrophages and fibroblats}
groupSize <- as.numeric(table(cellchat.all@idents))
par(mfrow = c(1,2), xpd=TRUE)
netVisual_circle(cellchat.all@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Number of interactions", targets.use = 13:21, sources.use = 6:8, color.use = colclusterName)
netVisual_circle(cellchat.all@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Interaction weights/strength", targets.use = 13:21, sources.use = 6:8, color.use = colclusterName)
```


```{r sender receiver OSM}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("OSM")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)

pairLR <- extractEnrichedLR(cellchat.all, signaling = pathways.show, geneLR.return = FALSE)
LR.show <- pairLR[1,] # second in list shown
## Circle plot
netVisual_individual(cellchat.all, signaling = pathways.show, pairLR.use = LR.show,layout = "circle", color.use = colclusterName, weight.scale = T)

pairLR <- extractEnrichedLR(cellchat.all, signaling = pathways.show, geneLR.return = FALSE)
LR.show <- pairLR[2,] # second in list shown
## Circle plot
netVisual_individual(cellchat.all, signaling = pathways.show, pairLR.use = LR.show,layout = "circle", color.use = colclusterName, weight.scale = T)

netVisual_heatmap(cellchat.all, signaling = pathways.show, color.heatmap = "Reds", color.use = colclusterName)

##show all the significant interactions (L-R pairs) associated with certain signaling pathways
netVisual_chord_gene(cellchat.all,signaling = c("OSM"), color.use = colclusterName, show.legend = FALSE)

## show all the significant interactions (L-R pairs) associated with certain signaling pathways 
## between macrophages (6-8) and fibroblasts (13:18)
netVisual_bubble(cellchat.all, sources.use = c(6:8), targets.use = c(13:18), remove.isolate = FALSE, pathways.show)
#> Comparing communications on a single object

## Chord diagram
par(mfrow=c(1,1))
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "chord", color.use = colclusterName )
```

```{r sender receiver IL1}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("IL1")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```
```{r sender receiver TGFb}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("TGFb")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver THBS}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("THBS")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver IL6}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("IL6")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```
```{r sender receiver SPP1}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("SPP1")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver LIGHT}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("LIGHT")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver ACTIVIN}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("ACTIVIN")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver VISFATIN}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("VISFATIN")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver SELL}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("SELL")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver SEMA4}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("SEMA4")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver GDF}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("GDF")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver TNF}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("TNF")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

```{r sender receiver GRN}
##show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
#netVisual_bubble(cellchat.all, sources.use = c(10:12), targets.use = c(13:21), remove.isolate = FALSE)
pathways.show <- c("GRN")
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat.all, signaling = pathways.show, width = 8, height = 2.5, font.size = 10, color.heatmap = "Blues", color.use = colclusterName)

##Hierarchy plot
##Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = c(1:12) # a numeric vector. 
netVisual_aggregate(cellchat.all, signaling = pathways.show, vertex.receiver = vertex.receiver, layout = "hierarchy", color.use = colclusterName)
##circle plot with pathway
netVisual_aggregate(cellchat.all, signaling = pathways.show, layout = "circle", color.use = colclusterName)
##plot contributing interactions
netAnalysis_contribution(cellchat.all, signaling = pathways.show)
```

#####################################compare wt vs cko#######################################

```{r load cellchat object wt and cko}
cellchat.cko <- readRDS("/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_cko.rds")
cellchat.wt <- readRDS("/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_wt.rds")
```

load and merge CellChat objects
```{r merge CellChat object, eval=FALSE, include=TRUE}

##lift cell CellChat objects to same lables
#define labels 
cellchat.all@idents
cellchat.cko@idents
cellchat.wt@idents
#labels.new = c("a_ACTA2PRC1", "b_ACTA2PRC2","c_ACTA2PRC3","d_ACTA2PRC4","e_ACTA2PRC5","f_FDC","g_TRC1","h_TRC2","i_TRC3","j_PI161","k_PI162","l_PI163","m_naiveCD4","n_Tcm","o_Treg","p_Tfh","q_CTL","r_naiveBcells","s_GCBcells","t_MBCs","u_plasmaCells")
#cellchat.tonsillitis <- liftCellChat(cellchat.tonsillitis, labels.new)
#cellchat.adult <- liftCellChat(cellchat.adult, labels.new)

object.list <- list(wt = cellchat.wt, cko = cellchat.cko)
cellchat <- mergeCellChat(object.list, add.names = names(object.list))
```

```{r save nerged cellchat object, eval=FALSE, include=TRUE}
saveRDS(cellchat, file = "/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_merged.rds")
```

```{r load merged cellchat object}
cellchat <- readRDS("/Users/immbio/Desktop/Project/Angelina/Cxcl13Hpb/data/cellchat_merged.rds")
object.list <- list(wt = cellchat.wt, cko = cellchat.cko)
df.net.wt <- subsetCommunication(cellchat.wt)
df.net.cko <- subsetCommunication(cellchat.cko)
```

```{r ggplotr overall number and strenght}
#set color vector
colCond <- c("#202547","#BE3144")

gg1 <- compareInteractions(cellchat, show.legend = F, group = c(1,2), color.use = colCond,)
gg2 <- compareInteractions(cellchat, show.legend = F, group = c(1,2), measure = "weight", color.use = colCond)
gg1 + gg2
```

Differential number of interactions or interaction strength among different cell populations
The differential number of interactions or interaction strength in the cell-cell communication network between two datasets can be visualized using circle plot, where red(or blue) colored edges represent increased (or decreased
) signaling in the second dataset compared to the first one.
```{r differential number of interactions mph to fibroblasts}
##circleplot interaction numbers strenght in diff subsets
par(mfrow = c(1,2), xpd=TRUE)
netVisual_diffInteraction(cellchat, weight.scale = T, sources.use = 6:8, targets.use = 13:18, color.use = colclusterName, color.edge = c("#BE3144", "#202547"),  vertex.weight = groupSize)
netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight", sources.use = 6:8, targets.use = 13:18, color.use = colclusterName, color.edge = c("#BE3144", "#202547"),  vertex.weight = groupSize)
```

```{r differential number of interactions immune cells to fibroblasts}
par(mfrow = c(1,2), xpd=TRUE)
netVisual_diffInteraction(cellchat, weight.scale = T, color.use = colclusterName, sources.use = 1:12, targets.use = 13:21)
netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight", color.use = colclusterName, sources.use = 1:12, targets.use = 13:21)
```

```{r differential number of interactions fibroblasts to immune cells}
par(mfrow = c(1,2), xpd=TRUE)
netVisual_diffInteraction(cellchat, weight.scale = T, color.use = colclusterName, sources.use = 13:21, targets.use = 1:12)
netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight", color.use = colclusterName, sources.use = 13:21, targets.use = 1:12)
```


```{r signaling changes}
#Furthermore, we can identify the specific signaling changes of Inflam.DC and cDC1 between NL and LS. ## Identify signaling changes associated with one cell group
gg1 <- netAnalysis_signalingChanges_scatter(cellchat, idents.use = "f_Mph1")
gg2 <- netAnalysis_signalingChanges_scatter(cellchat, idents.use = "g_Mph2")
gg3 <- netAnalysis_signalingChanges_scatter(cellchat, idents.use = "h_Mph3")
patchwork::wrap_plots(plots = list(gg1))
patchwork::wrap_plots(plots = list(gg2))
patchwork::wrap_plots(plots = list(gg3))
```

```{r heatmap differential interaction strnght}
##heatmap interaction numbers strenght in diff subsets
gg1 <- netVisual_heatmap(cellchat, color.use = colclusterName)
##> Do heatmap based on a merged object
gg2 <- netVisual_heatmap(cellchat, measure = "weight", color.use = colclusterName)
##> Do heatmap based on a merged object
gg1 + gg2
```

```{r rel information flow}
#relative information flow 
gg1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE, color.use = colCond)
gg2 <- rankNet(cellchat, mode = "comparison", stacked = F, do.stat = TRUE, color.use = colCond)
gg1 + gg2
```

```{r rel information flow mph1 to fb}
#relative information flow between Mph (6:8) and fibroblasts (13-18)
gg1 <- rankNet(cellchat, mode = "comparison", do.stat = TRUE,stacked = T, sources.use = 6:8, targets.use = 13:18, color.use = colCond)
gg2 <- rankNet(cellchat, mode = "comparison", do.stat = TRUE,stacked = F, sources.use = 6:8, targets.use = 13:18, color.use = colCond)
gg1 + gg2
```
```{r compare OSM pathway}
##compare pathways bt. wt and cko
pathways.show <- c("OSM") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,2), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, color.use = colclusterName ,layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}
```

```{r bubble upregulated signlinag ligand-receptor pairs}
#Moreover, we can identify the upgulated (increased) and down-regulated (decreased) signaling ligand-receptor pairs in one dataset compared to the other dataset. This can be done by specifying max.dataset and min.dataset in the function netVisual_bubble. The increased signaling means these signaling have higher communication probability (strength) in one dataset compared to the other dataset.

gg1 <- netVisual_bubble(cellchat, sources.use = 6:8, targets.use = 13:18,  comparison = c(1, 2), max.dataset = 1, title.name = "increased in wt", angle.x = 45, remove.isolate = T)
#> Comparing communications on a merged object
gg2 <- netVisual_bubble(cellchat, sources.use = 6:8, targets.use = 13:18,  comparison = c(1, 2), max.dataset = 2, title.name = "increased in cko", angle.x = 45, remove.isolate = T)
#> Comparing communications on a merged object
gg1 + gg2
```


```{r up and down regulated interactions}
#The above method for identifying the upgulated and down-regulated signaling is perfomed by comparing the communication probability between two datasets for each L-R pair and each pair of cell groups. Alternative, we can identify the upgulated and down-regulated signaling ligand-receptor pairs based on the differential gene expression analysis. Specifically, we perform differential expression analysis between two biological conditions (i.e., NL and LS) for each cell group, and then obtain the upgulated and down-regulated signaling based on the fold change of ligands in the sender cells and receptors in the receiver cells. Such analysis can be done as follows.

# define a positive dataset, i.e., the dataset with positive fold change against the other dataset
pos.dataset = "wt"
# define a char name used for storing the results of differential expression analysis
features.name = pos.dataset
# perform differential expression analysis
cellchat <- identifyOverExpressedGenes(cellchat, group.dataset = "datasets", pos.dataset = pos.dataset, features.name = features.name, only.pos = FALSE, thresh.pc = 0.1, thresh.fc = 0.1, thresh.p = 1)
#> Use the joint cell labels from the merged CellChat object
# map the results of differential expression analysis onto the inferred cell-cell communications to easily manage/subset the ligand-receptor pairs of interest
net <- netMappingDEG(cellchat, features.name = features.name)
# extract the ligand-receptor pairs with upregulated ligands in wt
net.up <- subsetCommunication(cellchat, net = net, datasets = "wt",ligand.logFC = 0.2, receptor.logFC = NULL)
# extract the ligand-receptor pairs with upregulated ligands and upregulated recetptors in cko, i.e.,downregulated in wt
net.down <- subsetCommunication(cellchat, net = net, datasets = "cko",ligand.logFC = -0.1, receptor.logFC = -0.1)
#Since the signaling genes in the net.up and net.down might be complex with multi-subunits, we can do further deconvolution to obtain the individual signaling genes.

gene.up <- extractGeneSubsetFromPair(net.up, cellchat)
gene.down <- extractGeneSubsetFromPair(net.down, cellchat)

#We then visualize the upgulated and down-regulated signaling ligand-receptor pairs using bubble plot or chord diagram.
pairLR.use.up = net.up[, "interaction_name", drop = F]
gg1 <- netVisual_bubble(cellchat, pairLR.use = pairLR.use.up, sources.use = c(6:8), targets.use = c(13:18), comparison = c(1, 2),  angle.x = 90, remove.isolate = T,title.name = paste0("Up-regulated signaling in ", names(object.list)[2]))
#> Comparing communications on a merged object
pairLR.use.down = net.down[, "interaction_name", drop = F]
gg2 <- netVisual_bubble(cellchat, pairLR.use = pairLR.use.down, sources.use = c(6:8), targets.use = c(13:18), comparison = c(1, 2),  angle.x = 90, remove.isolate = T,title.name = paste0("Down-regulated signaling in ", names(object.list)[2]))
#> Comparing communications on a merged object
gg1 + gg2
```

## session info
```{r date and session info}
date()
sessionInfo()
```
